<div class="container">

<table style="width: 100%;"><tr>
<td>SigTree</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Perform statistical analysis of tightness for branches of a hierarchical cluster.
</h2>

<h3>Description</h3>

<p>Description: Given data from which a hierarchical tree is grown, compute measures of tightness for each branch, sample from the null distribution of these measures in the randomized data and compute the corresponding p-values.
</p>


<h3>Usage</h3>

<pre><code class="language-R">SigTree(myinput,mystat=c("all","fldc","bldc","fldcc","slb"),
        mymethod="complete",mymetric="euclidean",rand.fun=NA,
        by.block=NA,distrib=c("vanilla","Rparallel"),Ptail=TRUE,
        tailmethod=c("ML","MOM"),njobs=1,seed=NA,
        Nperm=ifelse(Ptail,1000,1000*nrow(myinput)),
        metric.args=list(),rand.args=list())
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>myinput</code></td>
<td>

<p>A matrix with rows corresponding to items to be clustered.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mystat</code></td>
<td>

<p>A character string specifying the measures of tightness to be computed and evaluated for significance of finding. See Details for the definitions of these measures. If <code>"all"</code> is chosen, all the first three measures, <code>"fldc"</code>, <code>"bldc"</code> and <code>"fldcc"</code>, and the corresponding p-values are computed. Otherwise, only the specified measure and its p-value are computed.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mymethod</code></td>
<td>

<p>A character string specifying the linkage method for hierarchical clustering, to be used by the <code>hclust</code> function. See <code>hclust</code> argument <code>method</code> for method options.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mymetric</code></td>
<td>

<p>A character string specifying the definition of dissimilarity (distance) among the data items. The options, in addition to those for the argument <code>method</code> of the <code>dist</code> functiton, are <code>"pearson"</code>,<code>"kendall"</code>, and <code>"spearman"</code>. If one of the latter three is chosen, the distances are computed as <code>as.dist(1 - </code> 
<br><code>cor(myinput))</code>, with the corresponding option for the <code>method</code> argument of the <code>cor</code> function.It can also be a character string specifying a user supplied dissimilarity (distance) function for <code>myinput</code>. See <code>details</code> and <code>examples</code> below for further explanation.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rand.fun</code></td>
<td>

<p>A character string specifying the permutation method to be applied to <code>myinput</code>. If NA(default), no permutation is performed. <code>"shuffle.column"</code> performs a random permutation independently within each column. With <code>"shuffle.block"</code>, a random permutation is performed independently within each block of columns, as specified by the <code>by.block</code> argument, and independently from the other blocks. It can also be a character string specifying a user supplied randomization function for <code>myinput</code>. See <code>details</code> and <code>examples</code> below for further explanation.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>by.block</code></td>
<td>

<p>A vector of the same length as the column dimension of <code>myinput</code>, to specify the blocking of columns of <code>myinput</code>. It is used in conjunction with <code>rand.fun</code>
<br><code>= "shuffle.block"</code>, and is ignored otherwise.  
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>distrib</code></td>
<td>

<p>One of <code>"vanilla", "Rparallel"</code> to specify the distributed computing option for the cluster assignment step. For <code>"vanilla"</code> (default)
no distributed computing is performed. For <code>"Rparallel"</code> the <code>parallel</code> package of <code>R</code> core is used for multi-core processing.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Ptail</code></td>
<td>

<p>Logical. If <code>Ptail</code> is TRUE(default), the Generalized Pareto Distribution is used to approximate the tail of the null distribution for each of the chosen measures. Otherwise, empirical p-values are computed directly from the corresponding samples.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tailmethod</code></td>
<td>

<p>A character string only needed to be specified if the <code>Ptail</code> is set to TRUE. For <code>"ML"</code> the parameters of the Generalized Pareto Distribution are estimated by likelihood maximization; for <code>"MOM"</code> they are estimated by the method of moments.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>njobs</code></td>
<td>

<p>A single integer specifying the number of worker jobs to create in case of distributed computation if <code>distrib = "Rparallel"</code>; ignored otherwise.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>

<p>An optional single integer value, to be used to set the random number generator seed (see <code>details</code>).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Nperm</code></td>
<td>

<p>A single integer specifying the size of a sample from the null distribution. See <code>details</code> for the default sample size.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>metric.args</code></td>
<td>

<p>Additional arguments for user-supplied dissimilarity (distance) function. See <code>details</code> and <code>examples</code> below for further explanation.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rand.args</code></td>
<td>

<p>Additional arguments for user-supplied randomization function. See <code>details</code> and <code>examples</code> below for further explanation.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>When <code>rand.fun</code> is set to the name of a user supplied randomization function, the first argument of that function should be set to <code>myinput</code>. See examples below.
</p>
<p>The measures of tightness are defined as follows. Denote a node in the tree by <code>a</code>, its sibling node by <code>b</code>, and their parent node by <code>p</code>. Let their respective geights be <code>ha</code>,<code>hb</code>,<code>hp</code>. Finally, let <code>Sx</code> mean that the measure <code>S</code> is computed for the node <code>x</code>. Then the definitions are
</p>
<p>fldc:
</p>
<p><code>Sa = (hp-ha)/hp</code>
</p>
<p>fldcc:
</p>
<p><code>Sa = (hp-(ha-hb)/2)/ha</code>
</p>
<p>bldc:
</p>
<p><code>Sp = (2*hp-ha-hb)/(2*hp)</code>
</p>
<p>slb:
</p>
<p><code>Sp = 2*hp-ha-hb</code>
</p>
<p>The first three measures test tightnss of all internal nodes at the same time, while <code>slb</code> only tests two-way split of input data. 
The <code>seed</code> argument is optional. Setting the seed ensures reproducibility of sampling from the null distribution.
</p>


<h3>Value</h3>

<p>If <code>rand.fun</code> is set to NA, the function returns a matrix whose rows correspond to the internal nodes of the tree and whose columns contain the tree structure as in the <code>merge</code> component of the class <code>hclust</code>; the <code>height</code> component of <code>hclust</code>;and columns tabulating the values of the measures of tightness specified by the <code>mystat</code> argument.
If <code>rand.fun</code> is set to a specific randomization method, an object of class <code>best</code> is returned. See <code>?best</code> for details.
</p>


<h3>Note</h3>


<p>If <code>mymetric</code> or <code>rand.fun</code> is a customized function, make sure you have read and write permission for your working directory.
</p>


<h3>Author(s)</h3>

<p>Guoli Sun, Alex Krasnitz
</p>


<h3>References</h3>

<p>Theo A. Knijnenburg, Lodewyk F. A. Wessels et al (2009)
Fewer permutations, more accurate P-values
</p>


<h3>See Also</h3>

<p><code>best</code>,<code>plot.best</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">####Each column is a gene expression profile for a case of leukemia. 
####Each case belongs to one of three subtypes.
data(leukemia)
#output only statistic table
mytable&lt;-SigTree(data.matrix(leukemia),mystat="all",
        mymethod="ward",mymetric="euclidean")
class(mytable)
## Not run: 
#use multicore processing to detect significant sub-clusters
mytable&lt;-SigTree(data.matrix(leukemia),mystat="all",
	mymethod="ward",mymetric="euclidean",rand.fun="shuffle.column",
	distrib="Rparallel",njobs=2,Ptail=TRUE,tailmethod="ML")
class(mytable)
####Each row after the 1st describes an item belonging to one of four subtypes. 
####Each column corresponds to a genomic location in one of 22 human chromosomes. 
####The 1st row contains the chromosome numbers.
data(T10)
#Perform randomization within each chromosome
chrom&lt;-as.numeric(T10[1,])
mydata&lt;-T10[-1,] 
mytable&lt;-SigTree(data.matrix(mydata),mystat="fldc",        
	mymethod="ward",mymetric="euclidean",rand.fun="shuffle.block",
	by.block=chrom,distrib="Rparallel",njobs=2,Ptail=TRUE,tailmethod="ML")
#Compute dissimilarity using a user-supplied distance function,
#and perform randomization using a user-supplied randomization function, 
#with additional arguments. 
#Both user-supplied functions are only useful as illustration.
mydist&lt;-function(x,y){return(dist(x)/y)}
myrand&lt;-function(x,z){return(apply(x+z,2,sample))}
mytable&lt;-SigTree(data.matrix(leukemia),mystat="fldc",
mymethod="ward",mymetric="mydist",rand.fun="myrand",
distrib="Rparallel",njobs=2,Ptail=TRUE,tailmethod="MOM",metric.args=list(3),
rand.args=list(2))

## End(Not run)
</code></pre>


</div>