<div class="container">

<table style="width: 100%;"><tr>
<td>T10</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Breast tumor single cells data
</h2>

<h3>Description</h3>

<p>This data set summarizes DNA copy number variation in 100 individual cancer cells harvested from a breast tumor. The cells belong to four subtypes, differing by ploidy.There are 47 Diploid+Pseudo-diploid, 24 Hypo-diploid, 4 Aneuploid B and 25 Aneuploid A cells. Their copy number profiles are summarized in terms of 354 amplification and deletion "cores", are computed by the CORE package.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(T10)</code></pre>


<h3>Format</h3>

<p>A data frame with 101 rows and 354 columns. Each column corresponds to a core.  The first row is integer and contains the chromosome number for each core. The remaining rows are numeric, with values between 0 and 1, and each represents a DNA copy number profile of a cell.
</p>


<h3>Details</h3>

<p>Please remove the first row before computing the distance matrix.
</p>


<h3>Source</h3>

<p>Alexander Krasnitz, Guoli Sun, Peter Andrews, and Michael Wigler(2013)
Target inference from collections of genomic intervals
</p>


<h3>References</h3>

<p>Alexander Krasnitz, Guoli Sun, Peter Andrews, and Michael Wigler(2013)
Target inference from collections of genomic intervals		
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(T10)
dim(T10)
</code></pre>


</div>