<div class="container">

<table style="width: 100%;"><tr>
<td>tca</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Fitting the TCA model</h2>

<h3>Description</h3>

<p>Fits the TCA model for an input matrix of features by observations that are coming from a mixture of <code>k</code> sources, under the assumption that each observation is a mixture of unique (unobserved) source-specific values (in each feature in the data). This function further allows to statistically test the effect of covariates on source-specific values. For example, in the context of tissue-level bulk DNA methylation data coming from a mixture of cell types (i.e. the input is methylation sites by individuals), <code>tca</code> allows to model the methylation of each individual as a mixture of cell-type-specific methylation levels that are unique to the individual. In addition, it allows to statistically test the effects of covariates and phenotypes on methylation at the cell-type level.
</p>


<h3>Usage</h3>

<pre><code class="language-R">tca(
  X,
  W,
  C1 = NULL,
  C1.map = NULL,
  C2 = NULL,
  refit_W = FALSE,
  refit_W.features = NULL,
  refit_W.sparsity = 500,
  refit_W.sd_threshold = 0.02,
  tau = NULL,
  vars.mle = FALSE,
  constrain_mu = FALSE,
  parallel = FALSE,
  num_cores = NULL,
  max_iters = 10,
  log_file = "TCA.log",
  debug = FALSE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>An <code>m</code> by <code>n</code> matrix of measurements of <code>m</code> features for <code>n</code> observations. Each column in <code>X</code> is assumed to be a mixture of <code>k</code> sources. Note that <code>X</code> must include row names and column names and that NA values are currently not supported. <code>X</code> should not include features that are constant across all observations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>W</code></td>
<td>
<p>An <code>n</code> by <code>k</code> matrix of weights - the weights of <code>k</code> sources for each of the <code>n</code> mixtures (observations). All the weights must be positive and each row - corresponding to the weights of a single observation - must sum up to 1. Note that <code>W</code> must include row names and column names and that NA values are currently not supported. In cases where only noisy estimates of <code>W</code> are available, <code>tca</code> can be set to re-estimate <code>W</code> (see <code>refit_W</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>C1</code></td>
<td>
<p>An <code>n</code> by <code>p1</code> design matrix of covariates that may affect the hidden source-specific values (possibly a different effect size in each source). Note that <code>C1</code> must include row names and column names and should not include an intercept term. NA values are currently not supported. Note that each covariate in <code>C1</code> results in <code>k</code> additional parameters in the model of each feature, therefore, in order to alleviate the possibility of model overfitting, it is advised to be mindful of the balance between the size of <code>C1</code> and the sample size in <code>X</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>C1.map</code></td>
<td>
<p>An <code>p1</code> by <code>k</code> matrix of 0/1 values, indicating for each of the <code>p1</code> covariates in <code>C1</code> whether to consider its potential effects on the values of each of the <code>k</code> sources (e.g., if position <code>i,j</code> in <code>C1.map</code> is 1 then the potential effect of the <code>i</code>-th covariate in <code>C1</code> on the j-th source will be considered). If <code>C1.map == NULL</code> then effects for all covariates in <code>C1</code> will be considered in each of the sources. Note that <code>C1.map</code> is available only if <code>constrain_mu == TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>C2</code></td>
<td>
<p>An <code>n</code> by <code>p2</code> design matrix of covariates that may affect the mixture (i.e. rather than directly the sources of the mixture; for example, variables that capture biases in the collection of the measurements). Note that <code>C2</code> must include row names and column names and should not include an intercept term. NA values are currently not supported.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>refit_W</code></td>
<td>
<p>A logical value indicating whether to re-estimate the input <code>W</code> under the TCA model.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>refit_W.features</code></td>
<td>
<p>A vector with the names of the features in <code>X</code> to consider when re-estimating <code>W</code> (i.e. when <code>refit_W == TRUE</code>). This is useful since in general only a subset of the features in <code>X</code> are expected to be highly informative for estimating <code>W</code>. If <code>refit_W.features == NULL</code> then the ReFACTor algorithm will be used for performing feature selection (see also <code>refit_W.sparsity, refit_W.sd_threshold</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>refit_W.sparsity</code></td>
<td>
<p>A numeric value indicating the number of features to select using the ReFACTor algorithm when re-estimating <code>W</code> (activated only if <code>refit_W == TRUE</code> and <code>refit_W.features == NULL</code>). Note that <code>refit_W.sparsity</code> must be lower or equal to the number of features in <code>X</code>. For more information, see the argument <code>sparsity</code> in refactor.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>refit_W.sd_threshold</code></td>
<td>
<p>A numeric value indicating a standard deviation threshold to be used for excluding low-variance features in <code>X</code> (activated only if <code>refit_W == TRUE</code> and <code>refit_W.features == NULL</code>). For more information, see the argument <code>sd_threshold</code> in refactor.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tau</code></td>
<td>
<p>A non-negative numeric value of the standard deviation of the measurement noise (i.e. the i.i.d. component of variation in the model). If <code>tau == NULL</code> then <code>tca</code> will estimate <code>tau</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vars.mle</code></td>
<td>
<p>A logical value indicating whether to use maximum likelihood estimation when learning the variances in the model. If <code>vars.mle == FALSE</code> then <code>tca</code> will use a non-negative least-squares optimization for learning the variances. In practice, both approaches appear to provide highly correlated models, however, setting <code>vars.mle == FALSE</code>allows a substantial speedup.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>constrain_mu</code></td>
<td>
<p>A logical value indicating whether to constrain the estimates of the mean parameters (i.e. <code class="reqn">\{\mu_{hj}\}</code>; see details below), in which case they will be constrained to the range of the values in <code>X</code>. Note that if <code>constrain_mu == TRUE</code> then <code>tca</code> will not output p-values for the effect sizes of the covariates in <code>C1</code> and in <code>C2</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parallel</code></td>
<td>
<p>A logical value indicating whether to use parallel computing (possible when using a multi-core machine).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num_cores</code></td>
<td>
<p>A numeric value indicating the number of cores to use (activated only if <code>parallel == TRUE</code>). If <code>num_cores == NULL</code> then all available cores except for one will be used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max_iters</code></td>
<td>
<p>A numeric value indicating the maximal number of iterations to use in the optimization of the TCA model (<code>max_iters</code> iterations will be used as long as the optimization does not converge earlier).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log_file</code></td>
<td>
<p>A path to an output log file. Note that if the file <code>log_file</code> already exists then logs will be appended to the end of the file. Set <code>log_file</code> to <code>NULL</code> to prevent output from being saved into a file; note that if <code>verbose == FALSE</code> then no output file will be generated regardless of the value of <code>log_file</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>debug</code></td>
<td>
<p>A logical value indicating whether to set the logger to a more detailed debug level; set <code>debug</code> to <code>TRUE</code> before reporting issues.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>A logical value indicating whether to print logs.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The TCA model assumes that the hidden source-specific values are random variables. Formally, denote by <code class="reqn">Z_{hj}^i</code> the source-specific value of observation <code class="reqn">i</code> in feature <code class="reqn">j</code> source <code class="reqn">h</code>, the TCA model assumes: </p>
<p style="text-align: center;"><code class="reqn">Z_{hj}^i \sim N(\mu_{hj},\sigma_{hj}^2)</code>
</p>
<p> where <code class="reqn">\mu_{hj},\sigma_{hj}</code> represent the mean and standard deviation that are specific to feature <code class="reqn">j</code>, source <code class="reqn">h</code>. The model further assumes that the observed value of observation <code class="reqn">i</code> in feature <code class="reqn">j</code> is a mixture of <code class="reqn">k</code> different sources: </p>
<p style="text-align: center;"><code class="reqn">X_{ji} =  \sum_{h=1}^k W_{ih}Z_{hj}^i + \epsilon_{ji}</code>
</p>
<p> where <code class="reqn">W_{ih}</code> is the non-negative proportion of source <code class="reqn">h</code> in the mixture of observation <code class="reqn">i</code> such that <code class="reqn">\sum_{h=1}^kW_{ih} = 1</code>, and <code class="reqn">\epsilon_{ji} \sim N(0,\tau^2)</code> is an i.i.d. component of variation that models measurement noise. Note that the mixture proportions in <code class="reqn">W</code> are, in general, unique for each individual, therefore each entry in <code class="reqn">X</code> is coming from a unique distribution (i.e. a different mean and a different variance).
</p>
<p>In cases where the true <code>W</code> is unknown, <code>tca</code> can be provided with noisy estimates of <code>W</code> and then re-estimate <code>W</code> as part of the optimization procedure (see argument <code>refit_W</code>). These initial estimates should not be random but rather capture the information in <code>W</code> to some extent. When the argument <code>refit_W</code> is used, it is typically the case that only a subset of the features should be used for re-estimating <code>W</code>. Therefore, when re-estimating <code>W</code>, <code>tca</code> performs feature selection using the ReFACTor algorithm; alternatively, it can also be provided with a user-specified list of features to be used in the re-estimation, assuming that such list of features that are most informative for estimating <code class="reqn">W</code> exist (see argument <code>refit_W.features</code>).
</p>
<p>Covariates that systematically affect the source-specific values <code class="reqn">Z_{hj}^i</code> can be further considered (see argument <code>C1</code>). In that case, we assume: </p>
<p style="text-align: center;"><code class="reqn">Z_{hj}^i \sim N(\mu_{hj}+c^{(1)}_i \gamma_j^h,\sigma_{hj}^2)</code>
</p>
<p> where <code class="reqn">c^{(1)}_i</code> and <code class="reqn">\gamma_j^h</code> correspond to the <code class="reqn">p_1</code> covariate values of observation <code class="reqn">i</code> (i.e. a row vector from <code>C1</code>) and their effect sizes, respectively.
</p>
<p>Covariates that systematically affect the mixture values <code class="reqn">X_{ji}</code>, such as variables that capture technical biases in the collection of the measurements, can also be considered (see argument <code>C2</code>). In that case, we assume: </p>
<p style="text-align: center;"><code class="reqn">X_{ji} = \sum_{h=1}^k W_{ih}Z_{hj}^i + c^{(2)}_i \delta_j + \epsilon_{ij}</code>
</p>
<p> where <code class="reqn">c^{(2)}_i</code> and <code class="reqn">\delta_j</code> correspond to the <code class="reqn">p_2</code> covariate values of observation <code class="reqn">i</code> (i.e. a row vector from <code>C2</code>) and their effect sizes, respectively.
</p>
<p>Since the standard deviation of <code class="reqn">X_{ji}</code> is specific to observation <code class="reqn">i</code> and feature <code class="reqn">j</code>, we can obtain p-values for the estimates of <code class="reqn">\gamma_j^h</code> and <code class="reqn">\delta_j</code> by dividing each observed data point <code class="reqn">x_{ji}</code> by its estimated standard deviation and calculating T-statistics under a standard linear regression framework.
</p>


<h3>Value</h3>

<p>A list with the estimated parameters of the model. This list can be then used as the input to other functions such as tcareg.
</p>
<table>
<tr style="vertical-align: top;">
<td><code>W</code></td>
<td>
<p>An <code>n</code> by <code>k</code> matrix of weights. If <code>refit_W == TRUE</code> then this is the re-estimated <code>W</code>, and otherwise this is the input <code>W</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mus_hat</code></td>
<td>
<p>An <code>m</code> by <code>k</code> matrix of estimates for the mean of each source in each feature.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigmas_hat</code></td>
<td>
<p>An <code>m</code> by <code>k</code> matrix of estimates for the standard deviation of each source in each feature.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tau_hat</code></td>
<td>
<p>An estimate of the standard deviation of the i.i.d. component of variation in <code>X</code>. If an input value was provided for <code>tau</code> (i.e. <code>tau != NULL</code>) then <code>tau_hat == tau</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gammas_hat</code></td>
<td>
<p>An <code>m</code> by <code>k*p1</code> matrix of the estimated effects of the <code>p1</code> covariates in <code>C1</code> on each of the <code>m</code> features in <code>X</code>, where the first <code>p1</code> columns are the source-specific effects of the <code>p1</code> covariates on the first source, the following <code>p1</code> columns are the source-specific effects on the second source and so on.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>deltas_hat</code></td>
<td>
<p>An <code>m</code> by <code>p2</code> matrix of the estimated effects of the <code>p2</code> covariates in <code>C2</code> on the mixture values of each of the <code>m</code> features in <code>X</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gammas_hat_pvals</code></td>
<td>
<p>An <code>m</code> by <code>k*p1</code> matrix of p-values for the estimates in <code>gammas_hat</code> (based on a T-test). Note that <code>gammas_hat_pvals</code> is set to NULL if <code>constrain_mu == TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gammas_hat_pvals.joint</code></td>
<td>
<p>An <code>m</code> by <code>p1</code> matrix of p-values for the joint effects (i.e. across all <code>k</code> sources) of each of the <code>p1</code> covariates in <code>C1</code> on each of the <code>m</code> features in <code>X</code> (based on a partial F-test). In other words, these are p-values for the combined statistical effects of each one of the <code>p1</code> covariates on each of the <code>m</code> features under the TCA model. Note that <code>gammas_hat_pvals.joint</code> is set to NULL if <code>constrain_mu == TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>deltas_hat_pvals</code></td>
<td>
<p>An <code>m</code> by <code>p2</code> matrix of p-values for the estimates in <code>deltas_hat</code> (based on a T-test). Note that <code>deltas_hat_pvals</code> is set to NULL if <code>constrain_mu == TRUE</code>.</p>
</td>
</tr>
</table>
<h3>References</h3>

<p>Rahmani E, Schweiger R, Rhead B, Criswell LA, Barcellos LF, Eskin E, Rosset S, Sankararaman S, Halperin E. Cell-type-specific resolution epigenetics without the need for cell sorting or single-cell biology. Nature Communications 2019.
</p>
<p>Rahmani E, Zaitlen N, Baran Y, Eng C, Hu D, Galanter J, Oh S, Burchard EG, Eskin E, Zou J, Halperin E. Sparse PCA corrects for cell type heterogeneity in epigenome-wide association studies. Nature Methods 2016.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data &lt;- test_data(100, 20, 3, 1, 1, 0.01)
tca.mdl &lt;- tca(X = data$X, W = data$W, C1 = data$C1, C2 = data$C2)

</code></pre>


</div>