<div class="container">

<table style="width: 100%;"><tr>
<td>tcga</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Biological data</h2>

<h3>Description</h3>

<p>A list of biological data: RNASeq data, phenotypic
metadata and cell abundance.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(tcga)
</code></pre>


<h3>Details</h3>


<ul>
<li> <p><code>genes</code>: RNASeq from The Cancer Genome Atlas (TCGA) database.
</p>
</li>
<li> <p><code>phenotypes</code>: Metadata from the TCGA database containing
sample ID, sample type ID, sample type and primary disease.
</p>
</li>
<li> <p><code>cells</code>: Abundance estimates of cell types
</p>
</li>
</ul>
<h3>Note</h3>

<p>Subset of thirty samples of invasive breast carcinoma data from primary
tumor tissue. The cell type data are from a subset generated by the
Cibersort_ABS algorithm (<a href="https://cibersortx.stanford.edu">https://cibersortx.stanford.edu</a>).
For the complete dataset, please use:
<br><br><code>path &lt;- system.file("extdata", package = "tcgaViz")</code>
<br><code>load(file.path(path, "tcga.rda"))</code>
</p>


<h3>Source</h3>


<ul>
<li>
<p> dataset: gene expression RNAseq - Batch effects normalized mRNA data
</p>
</li>
<li>
<p> hub: <a href="https://pancanatlas.xenahubs.net">https://pancanatlas.xenahubs.net</a>
</p>
</li>
<li>
<p> cohort: TCGA Pan-Cancer (PANCAN)
</p>
</li>
<li>
<p> dataset ID: EB++AdjustPANCAN_IlluminaHiSeq_RNASeqV2.geneExp.xena
</p>
</li>
<li>
<p> download: <a href="https://tcga-pancan-atlas-hub.s3.us-east-1.amazonaws.com/download/EB%2B%2BAdjustPANCAN_IlluminaHiSeq_RNASeqV2.geneExp.xena.gz">https://tcga-pancan-atlas-hub.s3.us-east-1.amazonaws.com/download/EB%2B%2BAdjustPANCAN_IlluminaHiSeq_RNASeqV2.geneExp.xena.gz</a> (full metadata)
</p>
</li>
<li>
<p> samples: 11060
</p>
</li>
<li>
<p> version: 2016-12-29
</p>
</li>
<li>
<p> type of data: gene expression RNAseq
</p>
</li>
<li>
<p> unit: log2(norm_value+1)
</p>
</li>
<li>
<p> raw data: <a href="https://www.synapse.org/#!Synapse:syn4976369.3">https://www.synapse.org/#!Synapse:syn4976369.3</a>
</p>
</li>
<li>
<p> input data format: ROWs (identifiers) x COLUMNs (samples) (i.e. genomicMatrix)
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">data(tcga)
(df &lt;- convert2biodata(
    algorithm = "Cibersort_ABS",
    disease = "breast invasive carcinoma",
    tissue = "Primary Tumor",
    gene_x = "ICOS"
))
(stats &lt;- calculate_pvalue(df))

plot(df, stats = stats)

</code></pre>


</div>