<div class="container">

<table style="width: 100%;"><tr>
<td>plot.biodata</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Distribution plot</h2>

<h3>Description</h3>

<p>Distribution plot of cell subtypes according to the expression level (high or
low) of a selected gene.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'biodata'
plot(
  x,
  type = "violin",
  dots = FALSE,
  title = NULL,
  xlab = NULL,
  ylab = NULL,
  stats = NULL,
  draw = TRUE,
  axis.text.x = element_text(size = 10),
  axis.text.y = element_text(size = 8),
  cex.lab = 12,
  cex.main = 16,
  col = (scales::hue_pal())(length(unique(x$cell_type))),
  axis.title.x = element_text(size = cex.lab, face = "bold.italic", vjust = -0.5),
  axis.title.y = element_text(size = cex.lab, face = "bold.italic", vjust = -0.5),
  plot.title = element_text(size = cex.main, face = "bold", vjust = 1, hjust = 0.5),
  plot.margin = unit(c(0, 0, 0, -0.5), "cm"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>object from <code>convert2biodata()</code> for a dataframe containing
columns named high (logical), cell_type (factor) and value (float).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>character for the type of plot to be chosen among "violin"
or "boxplot".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dots</code></td>
<td>
<p>boolean to add all points to the graph.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title</code></td>
<td>
<p>character for the title of the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlab</code></td>
<td>
<p>character for the name of the X axis label.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylab</code></td>
<td>
<p>character for the name of the Y axis label.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>stats</code></td>
<td>
<p>object from <code>calculate_pvalue()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>draw</code></td>
<td>
<p>bolean to plot the graph.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>axis.text.x</code></td>
<td>
<p>tick labels along axes (<code>element_text()</code>). Specify all axis tick labels (<code>axis.text</code>),
tick labels by plane (using <code>axis.text.x</code> or <code>axis.text.y</code>), or individually
for each axis (using <code>axis.text.x.bottom</code>, <code>axis.text.x.top</code>,
<code>axis.text.y.left</code>, <code>axis.text.y.right</code>). <code style="white-space: pre;">⁠axis.text.*.*⁠</code> inherits from
<code style="white-space: pre;">⁠axis.text.*⁠</code> which inherits from <code>axis.text</code>, which in turn inherits
from <code>text</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>axis.text.y</code></td>
<td>
<p>tick labels along axes (<code>element_text()</code>). Specify all axis tick labels (<code>axis.text</code>),
tick labels by plane (using <code>axis.text.x</code> or <code>axis.text.y</code>), or individually
for each axis (using <code>axis.text.x.bottom</code>, <code>axis.text.x.top</code>,
<code>axis.text.y.left</code>, <code>axis.text.y.right</code>). <code style="white-space: pre;">⁠axis.text.*.*⁠</code> inherits from
<code style="white-space: pre;">⁠axis.text.*⁠</code> which inherits from <code>axis.text</code>, which in turn inherits
from <code>text</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.lab</code></td>
<td>
<p>numerical value giving the amount by which x and y plotting
labels should be magnified relative to the default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cex.main</code></td>
<td>
<p>numerical value giving the amount by which main plotting
title should be magnified relative to the default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col</code></td>
<td>
<p>character for the specification for the default plotting color.
See section 'Color Specification' in <code>graphics::par()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>axis.title.x</code></td>
<td>
<p>labels of axes (<code>element_text()</code>). Specify all axes' labels (<code>axis.title</code>),
labels by plane (using <code>axis.title.x</code> or <code>axis.title.y</code>), or individually
for each axis (using <code>axis.title.x.bottom</code>, <code>axis.title.x.top</code>,
<code>axis.title.y.left</code>, <code>axis.title.y.right</code>). <code style="white-space: pre;">⁠axis.title.*.*⁠</code> inherits from
<code style="white-space: pre;">⁠axis.title.*⁠</code> which inherits from <code>axis.title</code>, which in turn inherits
from <code>text</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>axis.title.y</code></td>
<td>
<p>labels of axes (<code>element_text()</code>). Specify all axes' labels (<code>axis.title</code>),
labels by plane (using <code>axis.title.x</code> or <code>axis.title.y</code>), or individually
for each axis (using <code>axis.title.x.bottom</code>, <code>axis.title.x.top</code>,
<code>axis.title.y.left</code>, <code>axis.title.y.right</code>). <code style="white-space: pre;">⁠axis.title.*.*⁠</code> inherits from
<code style="white-space: pre;">⁠axis.title.*⁠</code> which inherits from <code>axis.title</code>, which in turn inherits
from <code>text</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.title</code></td>
<td>
<p>plot title (text appearance) (<code>element_text()</code>; inherits
from <code>title</code>) left-aligned by default</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plot.margin</code></td>
<td>
<p>margin around entire plot (<code>unit</code> with the sizes of
the top, right, bottom, and left margins)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>arguments to pass to <code>ggplot2::theme()</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>No return value, called for side effects
</p>


<h3>Examples</h3>

<pre><code class="language-R">library("ggplot2")
data(tcga)
(df &lt;- convert2biodata(
    algorithm = "Cibersort_ABS",
    disease = "breast invasive carcinoma",
    tissue = "Primary Tumor",
    gene_x = "ICOS"
))

plot(df)
stats &lt;- calculate_pvalue(df)
plot(
    df,
    stats = stats,
    type = "boxplot",
    dots = TRUE,
    xlab = "Expression level of the 'ICOS' gene by cell type",
    ylab = "Percent of relative abundance\n(from the Cibersort_ABS algorithm)",
    title = "Differential analysis of tumor tissue immune cell type abundance
    based on RNASeq gene-level expression from The Cancer Genome Atlas
    (TCGA) database",
    axis.text.y = element_text(size = 8, hjust = 0.5),
    plot.title =  element_text(face = "bold", hjust = 0.5)
)

</code></pre>


</div>