<div class="container">

<table style="width: 100%;"><tr>
<td>omics</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Lung-cancer cell lines data in cancer cell line encyclopedia (CCLE) dataset</h2>

<h3>Description</h3>

<p>The omics data is a subset of the dataset provided by cancer cell line
encyclopedia (CCLE) project (Barretina et al., 2012;
<a href="https://sites.broadinstitute.org/ccle/">https://sites.broadinstitute.org/ccle/</a>).
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(omics)
</code></pre>


<h3>Format</h3>

<p>A list contains two objects:
</p>

<dl>
<dt>omics</dt>
<dd>
<p>a 3-dimensional array with size (3, 10, 68)</p>
</dd>
<dt>Y</dt>
<dd>
<p>a 68-length vector representing the response variable</p>
</dd>
</dl>
<h3>Details</h3>

<p>This data consists of one response variable and ten genes
evaluated under three different platforms. 
</p>
<p>The response variable measures the log-transformed activity area of taking 
Vandertanib, a drug targeting on EGFR gene for lung cancer. 
</p>
<p>The three platforms are DNA copy number variation (CNV), methylation and mRNA 
expression.
</p>
<p>Among the 10 genes, 7 of them (EGFR, EREG, HRAS, KRAS, PTPN11, STAT3, and
TGFA) are involved in the protein-protein interaction network of EGFR 
(<a href="https://string-db.org">https://string-db.org</a>) and the rest (ACTB, 
GAPDH, and PPIA) are arbitrarily chosen housekeeping genes and play the role
of negative control. 
</p>
<p>Detailed pre-processing procedure is available in Chang et al. (2021).
</p>


<h3>Source</h3>

<p><a href="https://string-db.org">https://string-db.org</a>
</p>


<h3>References</h3>

<p>Barretina, J., Caponigro, G., Stransky, N. et al. 
The Cancer Cell Line Encyclopedia enables predictive modelling of anticancer 
drug sensitivity. Nature 483, 603â€“607 (2012).
(<a href="https://www.nature.com/articles/nature11003">Link</a>)
</p>
<p>Sheng-Mao Chang, Meng Yang, Wenbin Lu, Yu-Jyun Huang, Yueyang Huang, Hung Hung, 
Jeffrey C Miecznikowski, Tzu-Pin Lu, Jung-Ying Tzeng, 
Gene-set integrative analysis of multi-omics data using tensor-based association test, 
Bioinformatics, 2021;, btab125,
(<a href="https://academic.oup.com/bioinformatics/advance-article-abstract/doi/10.1093/bioinformatics/btab125/6154849">Link</a>))
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(omics)
names(omics)
dim(omics$omics)
# 3 10 68
</code></pre>


</div>