<div class="container">

<table style="width: 100%;"><tr>
<td>rareGeneTest</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Rare Variant Tests</h2>

<h3>Description</h3>

<p>Runs tests for rare variants by running the "step-up" approach to choose the best set of variants (possibly signing them), and correcting by permutation. <code>rareGeneTest</code> tests a single gene, and also allows for other aggregation models described in Hoffmann et al.. <code>rarePathwayTest</code> provides an extension of this method, that does a step-up approach for inclusions of genes (for computional time). For larger genes, it may also be advantageous to run <code>rarePathwayTest</code> which will automatically chop large genes into subpieces and run step-up on those pieces.</p>


<h3>Usage</h3>

<pre><code class="language-R">rareGeneTest(genotype, phenotype,
  use_sign=TRUE, use_weight=TRUE, nperm=1000,
  binary=all(phenotype==0 | phenotype==1, na.rm=TRUE),
  strategy="step", thresh=1)
rarePathwayTest(genotype, genotype_gene, phenotype,
  use_sign=TRUE, use_weight=TRUE, nperm=1000,
  binary=all(phenotype==0 | phenotype==1, na.rm=TRUE),
  CUT=15)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>genotype</code></td>
<td>
<p>Matrix of individuals along the rows and genotypes along the columns. Each entry in the matrix should be coded <code>NA</code> for missing; and <code>0</code>, <code>1</code>, or <code>2</code> for the number of alleles a person has (must be bi-allelic).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phenotype</code></td>
<td>
<p>Vector of same length as the number of rows of the matrix. Either continuous, or dichotomous (case=<code>1</code>, control=<code>0</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use_sign</code></td>
<td>
<p>Whether to use a sign in the model for w, or just to weight all variants the same. Uses the sign of the estimated covariance between each marker and the trait to determine the sign of the marker.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use_weight</code></td>
<td>
<p>Whether to use a weight in the model for w (inverse variance of allele frequency in everyone (continuous) or just the controls (dichotomous)).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nperm</code></td>
<td>
<p>Number of permutations to run.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>binary</code></td>
<td>
<p>Is the phenotype dichotomous, or continuous? (TRUE/FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>strategy</code></td>
<td>
<p>‘step’ uses the step-up routine (default, recommended), ‘afreq’ tests all allele frequencies, and ‘thresh’ uses a fixed threshold.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>thresh</code></td>
<td>
<p>Allele frequency for inclusion, used only when <code>strategy</code>=‘thresh’.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotype_gene</code></td>
<td>
<p>For <code>rarePathwayTest</code>, this is a vector of length <code>ncol(genotype)</code>. This can be strings to indicate what gene each variant is in; e.g. 
<code>c("brca1",</code>
<code>"brca1",</code>
<code>"brca1",</code>
<code>"brca2",</code>
<code>"brca2")</code>
would indicate that the first three variants were in the first gene, and the last two variants are in the second gene. It can also be numeric.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>CUT</code></td>
<td>
<p>For <code>rarePathwayTest</code>, genes bigger than <code>CUT</code> will be chopped into subpieces of size smaller than <code>CUT</code> (for computational speed).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The methods here are as described in the Hoffmann et al. (PLoS ONE, to appear) paper. These methods are based on the idea of trying multiple models for rare variants, since prior information is generally not very accurate. The p-value is corrected for multiple comparisons by permutation.
</p>


<h3>References</h3>

<p>Hoffmann, TJ, Marini, NJ, and Witte, JS. Comprehensive approach to analyzing rare variants. PLoS ONE, 5(11): e13584. https://doi.org/10.1371/journal/pone.0013584.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## This inefficient (for clarity) code will simulate
##  a dataset in the correct format, and then run
##  the different approaches described here.

## Constants for data generation
SEED &lt;- 2
NCASE &lt;- NCONT &lt;- 500
NGENE &lt;- 6
FUNCTIONAL &lt;- 1:(NGENE/2)  ## Half are functional
BETA0 &lt;- -4
BETAG &lt;- log(2)

set.seed(SEED) ## Reproducible results

nonfunctional &lt;- setdiff(1:NGENE, FUNCTIONAL)
afreq &lt;- runif(NGENE, 0.001, 0.01)

expit &lt;- function(x)
  return(exp(x) / (1 + exp(x)))

gcase &lt;- matrix(0, nrow=NCASE, ncol=NGENE)
for(indiv in 1:NCASE){
  affected &lt;- FALSE
  while(!affected){ ## while not affected
    gcase[indiv, ] &lt;- rbinom(NGENE, 2, afreq)  ## draw up genotype
    affected &lt;- (expit(BETA0 + 
      BETAG*sum(gcase[indiv, FUNCTIONAL])) &gt; runif(1))
  }
}
cat("\n")

gcont &lt;- matrix(0, nrow=NCONT, ncol=NGENE)
for(indiv in 1:NCONT){
  unaffected &lt;- FALSE
  while(!unaffected){ ## while not unaffected
    gcont[indiv, ] &lt;- rbinom(NGENE, 2, afreq) ## draw up genotype
    unaffected &lt;- (1-expit(BETA0 +
      BETAG*sum(gcont[indiv, FUNCTIONAL])) &gt; runif(1))
  }
}
cat("\n")

cat("# Rare functional variants cases =", 
  sum(gcase[,FUNCTIONAL]), "\n")
cat("# Rare functional variants controls =",
  sum(gcont[,FUNCTIONAL]), "\n")
cat("# Rare non-functional variants cases =",
  sum(gcase[,nonfunctional]), "\n")
cat("# Rare non-functional variants controls =",
  sum(gcont[,nonfunctional]), "\n")

case &lt;- c(rep(1,NCASE), rep(0,NCONT))
genotype &lt;- rbind(gcase, gcont)

cat("P-value of the test:\n")
rareGeneTest(genotype, case)
</code></pre>


</div>