<div class="container">

<table style="width: 100%;"><tr>
<td>translate</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Convert DNA or RNA into proteins using genetic code</h2>

<h3>Description</h3>

<p>This function allows the user to input DNA or RNA sequences and
acquire sequences of corresponding proteins, where correspondence is encoded
in specified table.
</p>


<h3>Usage</h3>

<pre><code class="language-R">translate(x, table = 1, ...)

## S3 method for class 'sq_dna_bsc'
translate(x, table = 1, ..., NA_letter = getOption("tidysq_NA_letter"))

## S3 method for class 'sq_rna_bsc'
translate(x, table = 1, ..., NA_letter = getOption("tidysq_NA_letter"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>[<code>sq_dna_bsc</code> || <code>sq_rna_bsc</code>]<br>
An object this function is applied to.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>table</code></td>
<td>
<p>[<code>integer(1)</code>]<br>
The number of translation table used, as specified
<a href="https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi">here</a>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>further arguments to be passed from or to other methods.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>NA_letter</code></td>
<td>
<p>[<code>character(1)</code>]<br>
A string that is used to interpret and display <code>NA</code> value in the
context of <code>sq class</code>. Default value equals to
"<code>!</code>".</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>DNA and RNA sequences use combinations of three consecutive nucleic acids to
encode one of 22 amino acids. This encoding is called "genetic code".
</p>
<p><code>translate()</code> first splits passed DNA or RNA sequences into
three-letter chunks. Then searches the codon table for the entry where the
key is equal to the current chunk and the value is one letter that encodes
the corresponding protein. These resulting letters are then pasted into one
sequence for each input sequence.
</p>
<p>Due to how the tables works, <code>translate()</code> does not support inputting
sequences with extended alphabets, as ambiguous letters in most cases cannot
be translated into exactly one protein.
</p>
<p>Moreover, this function raises an error whenever input sequence contain
either "<code>-</code>" or <code>NA</code> value.
</p>


<h3>Value</h3>

<p>An object of <code>class sq</code> with <strong>ami_bsc</strong>
type.
</p>


<h3>See Also</h3>

<p><code>remove_ambiguous</code>, <code>substitute_letters</code> and
<code>typify</code> for necessary actions before using <code>translate()</code>
</p>
<p>Functions interpreting sq in biological context:
<code>%has%()</code>,
<code>complement()</code>,
<code>find_motifs()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">sq_dna &lt;- sq(c("TACTGGGCATGA", "CAGGTC", "TAGTCCTAG"), alphabet = "dna_bsc")
translate(sq_dna)

</code></pre>


</div>