<div class="container">

<table style="width: 100%;"><tr>
<td>data_exprs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Expression data from RNA-Seq study</h2>

<h3>Description</h3>

<p>This data was taken form a RNA-Seq study investigating the regulation of genes
in response to central nervous system inflammation.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data_exprs
</code></pre>


<h3>Format</h3>

<p>A data frame with 800 rows and 9 variables:
</p>

<dl>
<dt>ensembl_gene_id</dt>
<dd>
<p>Ensembl gene id</p>
</dd>
<dt>external_gene_name</dt>
<dd>
<p>Gene symbol</p>
</dd>
<dt>sample</dt>
<dd>
<p>Sample name</p>
</dd>
<dt>expression</dt>
<dd>
<p>Normalized RNA-Seq expression value</p>
</dd>
<dt>group</dt>
<dd>
<p>Experimental group</p>
</dd>
<dt>sample_type</dt>
<dd>
<p>Sample type. Either <code>input</code> or <code>IP</code>.</p>
</dd>
<dt>condition</dt>
<dd>
<p>Condition of sampling. Either <code>healthy</code> or <code>EAE</code>. </p>
</dd>
<dt>is_immune_gene</dt>
<dd>
<p>Gene is annotated as immune cell gene. Either <code>yes</code> or <code>no</code>.</p>
</dd>
<dt>direction</dt>
<dd>
<p>Direction of regulation. Either <code>up</code> or <code>down</code>. </p>
</dd>
</dl>
<h3>Source</h3>

<p><code>data_exprs</code> represents just a small subset of the data aquired in the study.
</p>
<p>More details about the study can be found here
</p>

<ul><li>
<p> Nature Neuroscience, <a href="https://www.nature.com/articles/s41593-019-0385-4">Bassoon proteinopathy drives neurodegeneration in multiple sclerosis</a>
</p>
</li></ul>
<p>The complete raw data can be downloaded here
</p>

<ul><li>
<p> Gene Expression Omnibus, study accession <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE104899">GSE104899</a>
</p>
</li></ul>
</div>