<div class="container">

<table style="width: 100%;"><tr>
<td>plotNovel</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Visualize evidence of novel V alleles</h2>

<h3>Description</h3>

<p><code>plotNovel</code> is be used to visualize the evidence of any novel V
alleles found using findNovelAlleles. It can also be used to
visualize the results for alleles that did
</p>


<h3>Usage</h3>

<pre><code class="language-R">plotNovel(
  data,
  novel_row,
  v_call = "v_call",
  j_call = "j_call",
  seq = "sequence_alignment",
  junction = "junction",
  junction_length = "junction_length",
  pos_range_max = NULL,
  ncol = 1,
  multiplot = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p><code>data.frame</code> containing repertoire data. See
findNovelAlleles for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>novel_row</code></td>
<td>
<p>single row from a data frame as output by
findNovelAlleles that contains a
polymorphism-containing germline allele.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>v_call</code></td>
<td>
<p>name of the column in <code>data</code> with V allele
calls. Default is <code>v_call</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>j_call</code></td>
<td>
<p>name of the column in <code>data</code> with J allele calls. 
Default is <code>j_call</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seq</code></td>
<td>
<p>name of the column in <code>data</code> with the 
aligned, IMGT-numbered, V(D)J nucleotide sequence.
Default is <code>sequence_alignment</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>junction</code></td>
<td>
<p>Junction region nucleotide sequence, which includes
the CDR3 and the two flanking conserved codons. Default
is <code>junction</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>junction_length</code></td>
<td>
<p>number of junction nucleotides in the junction sequence.
Default is <code>junction_length</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos_range_max</code></td>
<td>
<p>Name of the column in <code>data</code> with the ending
positions of the V alignment in the germline 
(usually <code>v_germline_end</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncol</code></td>
<td>
<p>number of columns to use when laying out the plots.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>multiplot</code></td>
<td>
<p>whether to return one single plot (<code>TRUE</code>) or a list 
with the three individual plots (<code>FALSE</code>).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The first panel in the plot shows, for all sequences which align to a particular 
germline allele, the mutation frequency at each postion along the aligned 
sequence as a function of the sequence-wide mutation count. Each line is a position.
Positions that contain polymorphisms (rather than somatic hypermutations) 
will exhibit a high apparent mutation frequency for a range of 
sequence-wide mutation counts. The positions are color coded as follows:
</p>

<ul>
<li>
<p>  red:    the position(s) pass(ess) the novel allele test 
</p>
</li>
<li>
<p>  yellow: the position(s) pass(ess) the y-intercept test but not
other tests
</p>
</li>
<li>
<p>  blue:   the position(s) didn't pass the y-intercept test and 
was(were) not further considered
</p>
</li>
</ul>
<p>The second panel shows the nucleotide usage at each of the polymorphic positions
as a function of sequence-wide mutation count. If no polymorphisms were identified,
the panel will show the mutation count.
</p>
<p>To avoid cases where a clonal expansion might lead to a false positive, TIgGER examines
the combinations of J gene and junction length among sequences which perfectly 
match the proposed germline allele. Clonally related sequences usually share 
the same V gene, J gene and junction length. Requiring the novel allele
to be found in different combinations of J gene and junction lengths
is a proxy for requiring it to be found in different clonal lineages.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Plot the evidence for the first (and only) novel allele in the example data
novel &lt;- selectNovel(SampleNovel)
plotNovel(AIRRDb, novel[1, ], v_call="v_call", j_call="j_call", 
          seq="sequence_alignment", junction="junction", junction_length="junction_length", 
          multiplot=TRUE)

</code></pre>


</div>