<div class="container">

<table style="width: 100%;"><tr>
<td>reassignAlleles</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Correct allele calls based on a personalized genotype</h2>

<h3>Description</h3>

<p><code>reassignAlleles</code> uses a subject-specific genotype to correct
correct preliminary allele assignments of a set of sequences derived
from a single subject.
</p>


<h3>Usage</h3>

<pre><code class="language-R">reassignAlleles(
  data,
  genotype_db,
  v_call = "v_call",
  seq = "sequence_alignment",
  method = "hamming",
  path = NA,
  keep_gene = c("gene", "family", "repertoire")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p><code>data.frame</code> containing V allele calls from a
single subject and the sample IMGT-gapped V(D)J sequences under
<code>seq</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotype_db</code></td>
<td>
<p>vector of named nucleotide germline sequences
matching the calls detailed in <code>allele_calls</code>
and personalized to the subject</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>v_call</code></td>
<td>
<p>name of the column in <code>data</code> with V allele
calls. Default is <code>v_call</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seq</code></td>
<td>
<p>name of the column in <code>data</code> with the 
aligned, IMGT-numbered, V(D)J nucleotide sequence.
Default is SEQUENCE_IMGT</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>method to use when realigning sequences to
the genotype_db sequences. Currently, only <code>"hammming"</code>
(for Hamming distance) is implemented.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>path</code></td>
<td>
<p>directory containing the tool used in the
realignment method, if needed. Hamming distance does
not require a path to a tool.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep_gene</code></td>
<td>
<p>string indicating if the gene (<code>"gene"</code>), 
family (<code>"family"</code>) or complete repertoire
(<code>"repertoire"</code>) assignments should be performed. 
Use of <code>"gene"</code> increases speed by minimizing required number of 
alignments, as gene level assignments will be maintained when possible.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>In order to save time, initial gene assignments are preserved and
the allele calls are chosen from among those provided in <code>genotype_db</code>,
based on a simple alignment to the sample sequence.
</p>


<h3>Value</h3>

<p>A modifed input <code>data.frame</code> containing the best allele call from 
among the sequences listed in <code>genotype_db</code> in the 
<code>v_call_genotyped</code> column.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Extract the database sequences that correspond to the genotype
genotype_db &lt;- genotypeFasta(SampleGenotype, SampleGermlineIGHV, novel=SampleNovel)

# Use the personlized genotype to determine corrected allele assignments
output_db &lt;- reassignAlleles(AIRRDb, genotype_db, v_call="v_call",
                             seq="sequence_alignment")

</code></pre>


</div>