<div class="container">

<table style="width: 100%;"><tr>
<td>tigger</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>tigger</h2>

<h3>Description</h3>

<p>Here we provide a <strong>T</strong>ool for <strong>I</strong>mmuno<strong>g</strong>lobulin
<strong>G</strong>enotype <strong>E</strong>lucidation via <strong>R</strong>ep-Seq (TIgGER). 
TIgGER inferrs the set of Ig alleles carried by an
individual (including any novel alleles) and then uses this set of alleles to
correct the initial assignments given to sample sequences by existing tools.
</p>


<h3>Details</h3>

<p>Immunoglobulin repertoire sequencing (AIRR-Seq, Rep-Seq) data is currently the
subject of much study. A key step in analyzing these data involves assigning
the closest known V(D)J germline alleles to the (often somatically mutated)
sample sequences using a tool such as IMGT/HighV-QUEST. However, if the
sample utilizes alleles not in the germline database used for alignment, this
step will fail. Additionally, this alignment has an associated error rate of
~5
mutations. The purpose of TIgGER is to address these issues.
</p>


<h3>Allele detection and genotyping</h3>


<ul>
<li>  <p>findNovelAlleles:       Detect novel alleles.
</p>
</li>
<li>  <p>plotNovel:              Plot evidence of novel alleles.
</p>
</li>
<li>  <p>inferGenotype:          Infer an Ig genotype using a frequency approach.
</p>
</li>
<li>  <p>inferGenotypeBayesian:  Infer an Ig genotype using a Bayesian approach.
</p>
</li>
<li>  <p>plotGenotype:           A colorful genotype visualization.
</p>
</li>
<li>  <p>genotypeFasta:          Convert a genotype to sequences.
</p>
</li>
<li>  <p>reassignAlleles:        Correct allele calls.
</p>
</li>
<li>  <p>generateEvidence:       Generate evidence for the genotype and 
allele detection inferrence.
</p>
</li>
</ul>
<h3>Mutation handling</h3>


<ul>
<li>  <p>getMutatedPositions:      Find mutation locations.
</p>
</li>
<li>  <p>getMutCount:              Find distance from germline.
</p>
</li>
<li>  <p>findUnmutatedCalls:       Subset unmutated sequences.
</p>
</li>
<li>  <p>getPopularMutationCount:  Find most common sequence's
mutation count.
</p>
</li>
<li>  <p>insertPolymorphisms:      Insert SNPs into a sequence.
</p>
</li>
</ul>
<h3>Input, output and formatting</h3>


<ul>
<li>  <p>readIgFasta:        Read a fasta file of Ig sequences.
</p>
</li>
<li>  <p>updateAlleleNames:  Correct outdated allele names.
</p>
</li>
<li>  <p>sortAlleles:        Sort allele names intelligently.
</p>
</li>
<li>  <p>cleanSeqs:          Standardize sequence format.
</p>
</li>
</ul>
<h3>References</h3>


<ol><li>
<p> Gadala-Maria, et al. (2015) Automated analysis of high-throughput B cell 
sequencing data reveals a high frequency of novel immunoglobulin V gene 
segment alleles. PNAS. 112(8):E862-70.
</p>
</li></ol>
</div>