<div class="container">

<table style="width: 100%;"><tr>
<td>inferGenotype</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Infer a subject-specific genotype using a frequency method</h2>

<h3>Description</h3>

<p><code>inferGenotype</code> infers an subject's genotype using a frequency method.
The genotype is inferred by finding the minimum number set of alleles that 
can explain the majority of each gene's calls. The most common allele of 
each gene is included in the genotype first, and the next most common allele 
is added until the desired fraction of alleles can be explained. In this 
way, mistaken allele calls (resulting from sequences which
by chance have been mutated to look like another allele) can be removed.
</p>


<h3>Usage</h3>

<pre><code class="language-R">inferGenotype(
  data,
  germline_db = NA,
  novel = NA,
  v_call = "v_call",
  seq = "sequence_alignment",
  fraction_to_explain = 0.875,
  gene_cutoff = 1e-04,
  find_unmutated = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p><code>data.frame</code> containing V allele
calls from a single subject.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>germline_db</code></td>
<td>
<p>named vector of sequences containing the
germline sequences named in
<code>allele_calls</code>. Only required if
<code>find_unmutated</code> is <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>novel</code></td>
<td>
<p>optional <code>data.frame</code> of the type
novel returned by
findNovelAlleles containing
germline sequences that will be utilized if
<code>find_unmutated</code> is <code>TRUE</code>. See
Details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>v_call</code></td>
<td>
<p>column in <code>data</code> with V allele calls.
Default is <code>"v_call"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seq</code></td>
<td>
<p>name of the column in <code>data</code> with the 
aligned, IMGT-numbered, V(D)J nucleotide sequence.
Default is <code>sequence_alignment</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fraction_to_explain</code></td>
<td>
<p>the portion of each gene that must be
explained by the alleles that will be included
in the genotype.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene_cutoff</code></td>
<td>
<p>either a number of sequences or a fraction of
the length of <code>allele_calls</code> denoting the
minimum number of times a gene must be
observed in <code>allele_calls</code> to be included
in the genotype.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>find_unmutated</code></td>
<td>
<p>if <code>TRUE</code>, use <code>germline_db</code> to
find which samples are unmutated. Not needed
if <code>allele_calls</code> only represent
unmutated samples.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Allele calls representing cases where multiple alleles have been
assigned to a single sample sequence are rare among unmutated
sequences but may result if nucleotides for certain positions are
not available. Calls containing multiple alleles are treated as
belonging to all groups. If <code>novel</code> is provided, all
sequences that are assigned to the same starting allele as any
novel germline allele will have the novel germline allele appended
to their assignent prior to searching for unmutated sequences.
</p>


<h3>Value</h3>

<p>A <code>data.frame</code> of alleles denoting the genotype of the subject containing 
the following columns:
</p>

<ul>
<li> <p><code>gene</code>: The gene name without allele.
</p>
</li>
<li> <p><code>alleles</code>: Comma separated list of alleles for the given <code>gene</code>.
</p>
</li>
<li> <p><code>counts</code>: Comma separated list of observed sequences for each 
corresponding allele in the <code>alleles</code> list.
</p>
</li>
<li> <p><code>total</code>: The total count of observed sequences for the given <code>gene</code>.
</p>
</li>
<li> <p><code>note</code>: Any comments on the inferrence.
</p>
</li>
</ul>
<h3>Note</h3>

<p>This method works best with data derived from blood, where a large
portion of sequences are expected to be unmutated. Ideally, there
should be hundreds of allele calls per gene in the input.
</p>


<h3>See Also</h3>

<p>plotGenotype for a colorful visualization and
genotypeFasta to convert the genotype to nucleotide sequences.
See inferGenotypeBayesian to infer a subject-specific genotype 
using a Bayesian approach.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Infer IGHV genotype, using only unmutated sequences, including novel alleles
inferGenotype(AIRRDb, germline_db=SampleGermlineIGHV, novel=SampleNovel,
              find_unmutated=TRUE)

</code></pre>


</div>