<div class="container">

<table style="width: 100%;"><tr>
<td>getPopularMutationCount</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Find mutation counts for frequency sequences</h2>

<h3>Description</h3>

<p><code>getPopularMutationCount</code> determines which sequences occur frequently
for each V gene and returns the mutation count of those sequences.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getPopularMutationCount(
  data,
  germline_db,
  v_call = "v_call",
  seq = "sequence_alignment",
  gene_min = 0.001,
  seq_min = 50,
  seq_p_of_max = 1/8,
  full_return = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p><code>data.frame</code> in the Change-O format. See
findNovelAlleles for a list of required
columns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>germline_db</code></td>
<td>
<p>named list of IMGT-gapped germline sequences.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>v_call</code></td>
<td>
<p>name of the column in <code>data</code> with V allele calls. 
Default is <code>v_call</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seq</code></td>
<td>
<p>name of the column in <code>data</code> with the 
aligned, IMGT-numbered, V(D)J nucleotide sequence.
Default is <code>sequence_alignment</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene_min</code></td>
<td>
<p>portion of all unique sequences a gene must
constitute to avoid exclusion.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seq_min</code></td>
<td>
<p>number of copies of the V that must be present for
to avoid exclusion.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seq_p_of_max</code></td>
<td>
<p>ror each gene, the fraction of the most common V sequence
count that a sequence must meet to avoid exclusion.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>full_return</code></td>
<td>
<p>if <code>TRUE</code>, will return all <code>data</code> columns and
will include sequences with mutation count &lt; 1.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A data frame of genes that have a frequent sequence mutation count
above 1.
</p>


<h3>See Also</h3>

<p>getMutatedPositions can be used to find which positions
of a set of sequences are mutated.
</p>


<h3>Examples</h3>

<pre><code class="language-R">getPopularMutationCount(AIRRDb, SampleGermlineIGHV)

</code></pre>


</div>