<div class="container">

<table style="width: 100%;"><tr>
<td>findNovelAlleles</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Find novel alleles from repertoire sequencing data</h2>

<h3>Description</h3>

<p><code>findNovelAlleles</code> analyzes mutation patterns in sequences thought to
align to each germline allele in order to determine which positions
might be polymorphic.
</p>


<h3>Usage</h3>

<pre><code class="language-R">findNovelAlleles(
  data,
  germline_db,
  v_call = "v_call",
  j_call = "j_call",
  seq = "sequence_alignment",
  junction = "junction",
  junction_length = "junction_length",
  germline_min = 200,
  min_seqs = 50,
  auto_mutrange = TRUE,
  mut_range = 1:10,
  pos_range = 1:312,
  pos_range_max = NULL,
  y_intercept = 0.125,
  alpha = 0.05,
  j_max = 0.15,
  min_frac = 0.75,
  nproc = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p><code>data.frame</code> containing repertoire data. See details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>germline_db</code></td>
<td>
<p>vector of named nucleotide germline sequences
matching the V calls in <code>data</code>. These should be 
the gapped reference germlines used to make the V calls.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>v_call</code></td>
<td>
<p>name of the column in <code>data</code> with V allele calls. 
Default is <code>v_call</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>j_call</code></td>
<td>
<p>name of the column in <code>data</code> with J allele calls. 
Default is <code>j_call</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seq</code></td>
<td>
<p>name of the column in <code>data</code> with the 
aligned, IMGT-numbered, V(D)J nucleotide sequence.
Default is <code>sequence_alignment</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>junction</code></td>
<td>
<p>Junction region nucleotide sequence, which includes
the CDR3 and the two flanking conserved codons. Default
is <code>junction</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>junction_length</code></td>
<td>
<p>Number of junction nucleotides in the junction sequence.
Default is <code>junction_length</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>germline_min</code></td>
<td>
<p>the minimum number of sequences that must have a
particular germline allele call for the allele to
be analyzed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_seqs</code></td>
<td>
<p>minimum number of total sequences (within the
desired mutational range and nucleotide range)
required for the samples to be considered.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>auto_mutrange</code></td>
<td>
<p>if <code>TRUE</code>, the algorithm will attempt to
determine the appropriate mutation range
automatically using the mutation count of the most
common sequence assigned to each allele analyzed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mut_range</code></td>
<td>
<p>range of mutations that samples may carry and
be considered by the algorithm.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos_range</code></td>
<td>
<p>range of IMGT-numbered positions that should be
considered by the algorithm.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos_range_max</code></td>
<td>
<p>Name of the column in <code>data</code> with the ending
positions of the V alignment in the germline 
(usually <code>v_germline_end</code>). The end of the alignment will
be used to limit the range of positions to be 
considered to count mutations. With <code>NULL</code> all 
positions in the IMGT V region will be considered. In
this case, in sequences where the V was trimmed 
on the 3', mutated nucleotides could include nucleotides
from the CDR3.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y_intercept</code></td>
<td>
<p>y-intercept threshold above which positions should be
considered potentially polymorphic.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>alpha value used for determining whether the 
fit y-intercept is greater than the <code>y_intercept</code>
threshold.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>j_max</code></td>
<td>
<p>maximum fraction of sequences perfectly aligning
to a potential novel allele that are allowed to
utilize to a particular combination of junction
length and J gene. The closer to 1, the less strict 
the filter for junction length and J gene diversity
will be.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_frac</code></td>
<td>
<p>minimum fraction of sequences that must have
usable nucleotides in a given position for that
position to considered.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nproc</code></td>
<td>
<p>number of processors to use.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The TIgGER allele-finding algorithm, briefly, works as follows:
Mutations are determined through comparison to the provided germline.
Mutation frequency at each *position* is determined as a function of
*sequence-wide* mutation counts. Polymorphic positions exhibit a high
mutation frequency despite sequence-wide mutation count. False positive of
potential novel alleles resulting from clonally-related sequences are guarded
against by ensuring that sequences perfectly matching the potential novel
allele utilize a wide range of combinations of J gene and junction length.
</p>


<h3>Value</h3>

<p>A <code>data.frame</code> with a row for each known allele analyzed.
Besides metadata on the the parameters used in the search, each row will have
either a note as to where the polymorphism-finding algorithm exited or a
nucleotide sequence for the predicted novel allele, along with columns providing
additional evidence.
</p>
<p>The output contains the following columns:
</p>

<ul>
<li> <p><code>germline_call</code>: The input (uncorrected) V call.
</p>
</li>
<li> <p><code>note</code>: Comments regarding the inferrence.
</p>
</li>
<li> <p><code>polymorphism_call</code>: The novel allele call.
</p>
</li>
<li> <p><code>nt_substitutions</code>: Mutations identified in the novel allele, relative
to the reference germline (<code>germline_call</code>)
</p>
</li>
<li> <p><code>novel_imgt</code>: The novel allele sequence.
</p>
</li>
<li> <p><code>novel_imgt_count</code>:  The number of times the sequence <code>novel_imgt</code> 
is found in the input data. Considers the subsequence of <code>novel_imgt</code> 
in the <code>pos_range</code>.
</p>
</li>
<li> <p><code>novel_imgt_unique_j</code>: Number of distinct J calls associated to <code>novel_imgt</code> 
in the input data. Considers the subsequence of <code>novel_imgt</code> in the <code>pos_range</code>.       
</p>
</li>
<li> <p><code>novel_imgt_unique_cdr3</code>: Number of distinct CDR3 sequences associated
with <code>novel_imgt</code> in the input data. Considers the subsequence of <code>novel_imgt</code> 
in the <code>pos_range</code>.                                              
</p>
</li>
<li> <p><code>perfect_match_count</code>: Final number of sequences retained to call the new 
allele. These are unique sequences that have V segments that perfectly match 
the predicted germline in the <code>pos_range</code>.
</p>
</li>
<li> <p><code>perfect_match_freq</code>: <code>perfect_match_count / germline_call_count</code>
</p>
</li>
<li> <p><code>germline_call_count</code>: The number of sequences with the <code>germline_call</code> 
in the input data that were initially considered for the analysis.
</p>
</li>
<li> <p><code>germline_call_freq</code>: The fraction of sequences with the <code>germline_call</code> 
in the input data initially considered for the analysis.              
</p>
</li>
<li> <p><code>germline_imgt</code>: Germline sequence for <code>germline_call</code>.
</p>
</li>
<li> <p><code>germline_imgt_count</code>: The number of times the <code>germline_imgt</code> 
sequence is found in the input data.
</p>
</li>
<li> <p><code>mut_min</code>: Minimum mutation considered by the algorithm.
</p>
</li>
<li> <p><code>mut_max</code>: Maximum mutation considered by the algorithm.
</p>
</li>
<li> <p><code>mut_pass_count</code>: Number of sequences in the mutation range.
</p>
</li>
<li> <p><code>pos_min</code>: First position of the sequence considered by the algorithm (IMGT numbering).
</p>
</li>
<li> <p><code>pos_max</code>: Last position of the sequence considered by the algorithm (IMGT numbering).
</p>
</li>
<li> <p><code>y_intercept</code>: The y-intercept above which positions were considered 
potentially polymorphic.
</p>
</li>
<li> <p><code>y_intercept_pass</code>: Number of positions that pass the <code>y_intercept</code> threshold.
</p>
</li>
<li> <p><code>snp_pass</code>: Number of sequences that pass the <code>y_intercept</code> threshold and are
within the desired nucleotide range (<code>min_seqs</code>).
</p>
</li>
<li> <p><code>unmutated_count</code>: Number of unmutated sequences.
</p>
</li>
<li> <p><code>unmutated_freq</code>: Number of unmutated sequences over <code>germline_imgt_count</code>.
</p>
</li>
<li> <p><code>unmutated_snp_j_gene_length_count</code>: Number of distinct combinations
of SNP, J gene, and junction length.     
</p>
</li>
<li> <p><code>snp_min_seqs_j_max_pass</code>: Number of SNPs that pass both the <code>min_seqs</code> 
and <code>j_max</code> thresholds.
</p>
</li>
<li> <p><code>alpha</code>: Significance threshold to be used when constructing the 
confidence interval for the y-intercept.
</p>
</li>
<li> <p><code>min_seqs</code>: Input <code>min_seqs</code>. The minimum number of total sequences 
(within the desired mutational range and nucleotide range) required 
for the samples to be considered.
</p>
</li>
<li> <p><code>j_max</code>: Input <code>j_max</code>. The maximum fraction of sequences perfectly 
aligning to a potential novel allele that are allowed to utilize to a particular 
combination of junction length and J gene.
</p>
</li>
<li> <p><code>min_frac</code>: Input <code>min_frac</code>. The minimum fraction of sequences that must
have usable nucleotides in a given position for that position to be considered.
</p>
</li>
</ul>
<p>The following comments can appear in the <code>note</code> column:
</p>

<ul>
<li> <p><em>Novel allele found</em>: A novel allele was detected.
</p>
</li>
<li> <p><em>Same as:</em>: The same novel allele sequence
has been identified multiple times. If this happens, the function
will also throw the message 'Duplicated polymorphism(s) found'.
</p>
</li>
<li> <p><em>Plurality sequence too rare</em>: No sequence is frequent enough to pass 
the J test (<code>j_max</code>).
</p>
</li>
<li> <p><em>A J-junction combination is too prevalent</em>: Not enough J diversity (<code>j_max</code>).
</p>
</li>
<li> <p><em>No positions pass y-intercept test</em>: No positions above <code>y_intercept</code>.
</p>
</li>
<li> <p><em>Insufficient sequences in desired mutational range</em>: 
<code>mut_range</code> and <code>pos_range</code>.
</p>
</li>
<li> <p><em>Not enough sequences</em>: Not enough sequences in the desired mutational 
range and nucleotide range (<code>min_seqs</code>).
</p>
</li>
<li> <p><em>No unmutated versions of novel allele found</em>: All observed variants of the 
allele are mutated.
</p>
</li>
</ul>
<h3>See Also</h3>

<p>selectNovel to filter the results to show only novel alleles.
plotNovel to visualize the data supporting any
novel alleles hypothesized to be present in the data and
inferGenotype and inferGenotypeBayesian to determine if the novel alleles are frequent
enought to be included in the subject's genotype.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# Note: In this example, with SampleGermlineIGHV,
# which contains reference germlines retrieved on August 2014,
# TIgGER finds the allele IGHV1-8*02_G234T. This allele
# was added to IMGT as IGHV1-8*03 on March 28, 2018.

# Find novel alleles and return relevant data
novel &lt;- findNovelAlleles(AIRRDb, SampleGermlineIGHV)
selectNovel(novel)


</code></pre>


</div>