<div class="container">

<table style="width: 100%;"><tr>
<td>generateEvidence</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generate evidence</h2>

<h3>Description</h3>

<p><code>generateEvidence</code> builds a table of evidence metrics for the final novel V 
allele detection and genotyping inferrences.
</p>


<h3>Usage</h3>

<pre><code class="language-R">generateEvidence(
  data,
  novel,
  genotype,
  genotype_db,
  germline_db,
  j_call = "j_call",
  junction = "junction",
  fields = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>a <code>data.frame</code> containing sequence data that has been
passed through reassignAlleles to correct the allele 
assignments.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>novel</code></td>
<td>
<p>the <code>data.frame</code> returned by findNovelAlleles.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotype</code></td>
<td>
<p>the <code>data.frame</code> of alleles generated with inferGenotype 
denoting the genotype of the subject.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genotype_db</code></td>
<td>
<p>a vector of named nucleotide germline sequences in the genotype.
Returned by genotypeFasta.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>germline_db</code></td>
<td>
<p>the original uncorrected germline database used to by
findNovelAlleles to identify novel alleles.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>j_call</code></td>
<td>
<p>name of the column in <code>data</code> with J allele calls. 
Default is <code>j_call</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>junction</code></td>
<td>
<p>Junction region nucleotide sequence, which includes
the CDR3 and the two flanking conserved codons. Default
is <code>junction</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fields</code></td>
<td>
<p>character vector of column names used to split the data to 
identify novel alleles, if any. If <code>NULL</code> then the data is 
not divided by grouping variables.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns the <code>genotype</code> input <code>data.frame</code> with the following additional columns 
providing supporting evidence for each inferred allele:
</p>

<ul>
<li> <p><code>field_id</code>: Data subset identifier, defined with the input paramter <code>fields</code>.
</p>
</li>
<li>
<p> A variable number of columns, specified with the input parameter <code>fields</code>.
</p>
</li>
<li> <p><code>polymorphism_call</code>: The novel allele call.
</p>
</li>
<li> <p><code>novel_imgt</code>: The novel allele sequence.
</p>
</li>
<li> <p><code>closest_reference</code>: The closest reference gene and allele in 
the <code>germline_db</code> database.
</p>
</li>
<li> <p><code>closest_reference_imgt</code>: Sequence of the closest reference gene and 
allele in the <code>germline_db</code> database.
</p>
</li>
<li> <p><code>germline_call</code>: The input (uncorrected) V call.
</p>
</li>
<li> <p><code>germline_imgt</code>: Germline sequence for <code>germline_call</code>.
</p>
</li>
<li> <p><code>nt_diff</code>: Number of nucleotides that differ between the new allele and
the closest reference (<code>closest_reference</code>) in the <code>germline_db</code> database.
</p>
</li>
<li> <p><code>nt_substitutions</code>: A comma separated list of specific nucleotide 
differences (e.g. <code>112G&gt;A</code>) in the novel allele.
</p>
</li>
<li> <p><code>aa_diff</code>: Number of amino acids that differ between the new allele and the closest 
reference (<code>closest_reference</code>) in the <code>germline_db</code> database.
</p>
</li>
<li> <p><code>aa_substitutions</code>: A comma separated list with specific amino acid 
differences (e.g. <code>96A&gt;N</code>) in the novel allele.
</p>
</li>
<li> <p><code>sequences</code>: Number of sequences unambiguosly assigned to this allele.
</p>
</li>
<li> <p><code>unmutated_sequences</code>: Number of records with the unmutated novel allele sequence.
</p>
</li>
<li> <p><code>unmutated_frequency</code>: Proportion of records with the unmutated novel allele 
sequence (<code>unmutated_sequences / sequences</code>).
</p>
</li>
<li> <p><code>allelic_percentage</code>: Percentage at which the (unmutated) allele is observed 
in the sequence dataset compared  to other (unmutated) alleles.
</p>
</li>
<li> <p><code>unique_js</code>: Number of unique J sequences found associated with the 
novel allele. The sequences are those who have been unambiguously assigned 
to the novel allelle (<code>polymorphism_call</code>).
</p>
</li>
<li> <p><code>unique_cdr3s</code>: Number of unique CDR3s associated with the inferred allele.
The sequences are those who have been unambiguously assigned to the 
novel allelle (polymorphism_call).
</p>
</li>
<li> <p><code>mut_min</code>: Minimum mutation considered by the algorithm.
</p>
</li>
<li> <p><code>mut_max</code>: Maximum mutation considered by the algorithm.
</p>
</li>
<li> <p><code>pos_min</code>: First position of the sequence considered by the algorithm (IMGT numbering).
</p>
</li>
<li> <p><code>pos_max</code>: Last position of the sequence considered by the algorithm (IMGT numbering).
</p>
</li>
<li> <p><code>y_intercept</code>: The y-intercept above which positions were considered 
potentially polymorphic.
</p>
</li>
<li> <p><code>alpha</code>: Significance threshold to be used when constructing the 
confidence interval for the y-intercept.
</p>
</li>
<li> <p><code>min_seqs</code>: Input <code>min_seqs</code>. The minimum number of total sequences 
(within the desired mutational range and nucleotide range) required 
for the samples to be considered.
</p>
</li>
<li> <p><code>j_max</code>: Input <code>j_max</code>. The maximum fraction of sequences perfectly 
aligning to a potential novel allele that are allowed to utilize to a particular 
combination of junction length and J gene.
</p>
</li>
<li> <p><code>min_frac</code>: Input <code>min_frac</code>. The minimum fraction of sequences that must
have usable nucleotides in a given position for that position to be considered.
</p>
</li>
<li> <p><code>note</code>: Comments regarding the novel allele inferrence.
</p>
</li>
</ul>
<h3>See Also</h3>

<p>See findNovelAlleles, inferGenotype and genotypeFasta 
for generating the required input.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# Generate input data
novel &lt;- findNovelAlleles(AIRRDb, SampleGermlineIGHV,
    v_call="v_call", j_call="j_call", junction="junction", 
    junction_length="junction_length", seq="sequence_alignment")
genotype &lt;- inferGenotype(AIRRDb, find_unmutated=TRUE, 
                          germline_db=SampleGermlineIGHV,
                          novel=novel,
                          v_call="v_call", seq="sequence_alignment")
genotype_db &lt;- genotypeFasta(genotype, SampleGermlineIGHV, novel)
data_db &lt;- reassignAlleles(AIRRDb, genotype_db, 
v_call="v_call", seq="sequence_alignment")

# Assemble evidence table
evidence &lt;- generateEvidence(data_db, novel, genotype, 
                             genotype_db, SampleGermlineIGHV,
                             j_call = "j_call", 
                             junction = "junction")


</code></pre>


</div>