<div class="container">

<table style="width: 100%;"><tr>
<td>Tm_NN</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate melting temperature using nearest neighbor thermodynamics</h2>

<h3>Description</h3>

<p>Calculate melting temperature using nearest neighbor thermodynamics
</p>


<h3>Usage</h3>

<pre><code class="language-R">Tm_NN(
  ntseq,
  ambiguous = FALSE,
  comSeq = NULL,
  shift = 0,
  nn_table = c("DNA_NN4", "DNA_NN1", "DNA_NN2", "DNA_NN3", "RNA_NN1", "RNA_NN2",
    "RNA_NN3", "R_DNA_NN1"),
  tmm_table = "DNA_TMM1",
  imm_table = "DNA_IMM1",
  de_table = c("DNA_DE1", "RNA_DE1"),
  dnac1 = 25,
  dnac2 = 25,
  selfcomp = FALSE,
  Na = 0,
  K = 0,
  Tris = 0,
  Mg = 0,
  dNTPs = 0,
  saltcorr = c("Schildkraut2010", "Wetmur1991", "SantaLucia1996", "SantaLucia1998-1",
    "SantaLucia1998-2", "Owczarzy2004", "Owczarzy2008"),
  DMSO = 0,
  fmd = 0,
  DMSOfactor = 0.75,
  fmdfactor = 0.65,
  fmdmethod = c("concentration", "molar"),
  outlist = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ntseq</code></td>
<td>
<p>Sequence (5' to 3') of one strand of the nucleic acid duplex
as string or vector of characters.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ambiguous</code></td>
<td>
<p>Ambiguous bases are taken into account to compute the G
and C content when ambiguous is TRUE.Default is FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>comSeq</code></td>
<td>
<p>Complementary sequence. The sequence of the template/target in 3'-&gt;5' direction</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>shift</code></td>
<td>
<p>Shift of the primer/probe sequence on the template/target sequence, default=0. 
for example: when shift=0, the first nucleotide base at 5' end of primer align to first one at 3' 
end of template. When shift=-1, the second nucleotide base at 5' end of primer align to first one at 3' 
end of template.
</p>
<p>When shift=1, the first nucleotide base at 5' end of primer align to second one at 3' end 
of template. The shift parameter is necessary to align  primer/probe and template/target 
if they have different lengths or if they should have dangling ends.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nn_table</code></td>
<td>
<p>Thermodynamic NN values, eight tables are implemented.
</p>
<p>For DNA/DNA hybridizations:
DNA_NN1,DNA_NN2,DNA_NN3,DNA_NN4
</p>
<p>For RNA/RNA hybridizations:
RNA_NN1,RNA_NN2,RNA_NN3
</p>
<p>For RNA/DNA hybridizations:
R_DNA_NN1</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tmm_table</code></td>
<td>
<p>Thermodynamic values for terminal mismatches. Default: DNA_TMM1</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>imm_table</code></td>
<td>
<p>Thermodynamic values for internal mismatches, may include insosine mismatches. Default: DNA_IMM1</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>de_table</code></td>
<td>
<p>Thermodynamic values for dangling ends. DNA_DE1(default) and RNA_DE1</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dnac1</code></td>
<td>
<p>Concentration of the higher concentrated strand [nM]. Typically this will
be the primer (for PCR) or the probe. Default=25.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dnac2</code></td>
<td>
<p>Concentration of the lower concentrated strand [nM].</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>selfcomp</code></td>
<td>
<p>Sequence self-complementary, default=False. If 'True' 
the primer is thought binding to itself, thus dnac2 is not considered.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Na</code></td>
<td>
<p>Millimolar concentration of Na, default is 0</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K</code></td>
<td>
<p>Millimolar concentration of K, default is 0</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Tris</code></td>
<td>
<p>Millimolar concentration of Tris, default is 0</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Mg</code></td>
<td>
<p>Millimolar concentration of Mg, default is 0</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dNTPs</code></td>
<td>
<p>Millimolar concentration of dNTPs, default is 0</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>saltcorr</code></td>
<td>
<p>Salt correction method should be chosen when provide 'userset' 
Options are "Schildkraut2010", "Wetmur1991","SantaLucia1996","SantaLucia1998-1",
"SantaLucia1998-2","Owczarzy2004","Owczarzy2008". Note that NA means no salt correction.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>DMSO</code></td>
<td>
<p>Percent DMSO</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fmd</code></td>
<td>
<p>Formamide concentration in percentage (fmdmethod="concentration") or molar (fmdmethod="molar").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>DMSOfactor</code></td>
<td>
<p>Coeffecient of Tm decreases per percent DMSO. Default=0.75 von Ahsen N (2001) &lt;PMID:11673362&gt;. Other published values are 0.5, 0.6 and 0.675.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fmdfactor</code></td>
<td>
<p>Coeffecient of Tm decrease per percent formamide. Default=0.65. Several papers report factors between 0.6 and 0.72.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fmdmethod</code></td>
<td>
<p>"concentration" method for formamide concentration in percentage and "molar" for formamide concentration in molar.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outlist</code></td>
<td>
<p>output a list of Tm and options or only Tm value, default is TRUE.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>DNA_NN1: Breslauer K J (1986) &lt;doi:10.1073/pnas.83.11.3746&gt;
</p>
<p>DNA_NN2: Sugimoto N (1996) &lt;doi:10.1093/nar/24.22.4501&gt;
</p>
<p>DNA_NN3: Allawi H (1998) &lt;doi:10.1093/nar/26.11.2694&gt;
</p>
<p>DNA_NN4: SantaLucia J (2004) &lt;doi:10.1146/annurev.biophys.32.110601.141800&gt;
</p>
<p>RNA_NN1: Freier S (1986) &lt;doi:10.1073/pnas.83.24.9373&gt;
</p>
<p>RNA_NN2: Xia T (1998) &lt;doi:10.1021/bi9809425&gt;
</p>
<p>RNA_NN3: Chen JL (2012) &lt;doi:10.1021/bi3002709&gt;
</p>
<p>R_DNA_NN1: Sugimoto N (1995)&lt;doi:10.1016/S0048-9697(98)00088-6&gt;
</p>
<p>DNA_TMM1: Bommarito S (2000)  &lt;doi:10.1093/nar/28.9.1929&gt;
</p>
<p>DNA_IMM1: Peyret N (1999) &lt;doi:10.1021/bi9825091&gt; &amp; Allawi H T (1997) &lt;doi:10.1021/bi962590c&gt; &amp; Santalucia N (2005) &lt;doi:10.1093/nar/gki918&gt;
</p>
<p>DNA_DE1: Bommarito S (2000) &lt;doi:10.1093/nar/28.9.1929&gt;
</p>
<p>RNA_DE1: Turner D H (2010) &lt;doi:10.1093/nar/gkp892&gt;
</p>


<h3>Author(s)</h3>

<p>Junhui Li
</p>


<h3>References</h3>

<p>Breslauer K J , Frank R , Blocker H , et al. Predicting DNA duplex stability from the base sequence.[J]. Proceedings of the National Academy of Sciences, 1986, 83(11):3746-3750.
</p>
<p>Sugimoto N , Nakano S , Yoneyama M , et al. Improved Thermodynamic Parameters and Helix Initiation Factor to Predict Stability of DNA Duplexes[J]. Nucleic Acids Research, 1996, 24(22):4501-5.
</p>
<p>Allawi, H. Thermodynamics of internal C.T mismatches in DNA[J]. Nucleic Acids Research, 1998, 26(11):2694-2701.
</p>
<p>Hicks L D , Santalucia J . The thermodynamics of DNA structural motifs.[J]. Annual Review of Biophysics &amp; Biomolecular Structure, 2004, 33(1):415-440.
</p>
<p>Freier S M , Kierzek R , Jaeger J A , et al. Improved free-energy parameters for predictions of RNA duplex stability.[J]. Proceedings of the National Academy of Sciences, 1986, 83(24):9373-9377.
</p>
<p>Xia T , Santalucia , J , Burkard M E , et al. Thermodynamic Parameters for an Expanded Nearest-Neighbor Model for Formation of RNA Duplexes with Watson-Crick Base Pairs,[J]. Biochemistry, 1998, 37(42):14719-14735.
</p>
<p>Chen J L , Dishler A L , Kennedy S D , et al. Testing the Nearest Neighbor Model for Canonical RNA Base Pairs: Revision of GU Parameters[J]. Biochemistry, 2012, 51(16):3508-3522.
</p>
<p>Bommarito S, Peyret N, Jr S L. Thermodynamic parameters for DNA sequences with dangling ends[J]. Nucleic Acids Research, 2000, 28(9):1929-1934.
</p>
<p>Turner D H , Mathews D H . NNDB: the nearest neighbor parameter database for predicting stability of nucleic acid secondary structure[J]. Nucleic Acids Research, 2010, 38(Database issue):D280-D282.
</p>
<p>Sugimoto N , Nakano S I , Katoh M , et al. Thermodynamic Parameters To Predict Stability of RNA/DNA Hybrid Duplexes[J]. Biochemistry, 1995, 34(35):11211-11216.
</p>
<p>Allawi H, SantaLucia J: Thermodynamics and NMR of internal G-T mismatches in DNA. Biochemistry 1997, 36:10581-10594.
</p>
<p>Santalucia N E W J . Nearest-neighbor thermodynamics of deoxyinosine pairs in DNA duplexes[J]. Nucleic Acids Research, 2005, 33(19):6258-67.
</p>
<p>Peyret N , Seneviratne P A , Allawi H T , et al. Nearest-Neighbor Thermodynamics and NMR of DNA Sequences with Internal A-A, C-C, G-G, and T-T Mismatches, [J]. Biochemistry, 1999, 38(12):3468-3477.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
ntseq &lt;- c("AAAATTTTTTTCCCCCCCCCCCCCCGGGGGGGGGGGGTGTGCGCTGC")
out &lt;- Tm_NN(ntseq,Na=50)
out
out$Options

</code></pre>


</div>