<div class="container">

<table style="width: 100%;"><tr>
<td>gt2refalt</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Two-letter genotype from VCF GT</h2>

<h3>Description</h3>

<p>Get a two-letter genotype from a VCF GT field. Current
implementation is quick and dirty, and only accepts 0/0, 0/1, or 1/1. Any
other input to gt will return a missing value.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gt2refalt(gt, ref, alt)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gt</code></td>
<td>
<p>The genotype field (must be 0/0, 0/1, or 1/1).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref</code></td>
<td>
<p>The reference allele.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alt</code></td>
<td>
<p>The alternate allele.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returnvalue
</p>


<h3>Examples</h3>

<pre><code class="language-R">gt2refalt(gt="0/0", ref="R", alt="A")
gt2refalt(gt="0/1", ref="R", alt="A")
gt2refalt(gt="1/1", ref="R", alt="A")
gt2refalt(gt="0/2", ref="R", alt="A")
gt2refalt(gt="./.", ref="R", alt="A")
</code></pre>


</div>