<div class="container">

<table style="width: 100%;"><tr>
<td>create_snpset</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create a dataframe that can be used as input for making effect plots</h2>

<h3>Description</h3>

<p><code>create_snpset()</code>
</p>
<p>This method is deprecated and will be removed in future versions. use <code>get_snpset</code> instead.
</p>


<h3>Usage</h3>

<pre><code class="language-R">create_snpset(
  df1,
  df2,
  thresh = 1e-08,
  protein_coding_only = TRUE,
  region_size = 1e+06,
  verbose = F
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>df1</code></td>
<td>
<p>The dataframe to extract the top snps from (with p-value below thresh)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>df2</code></td>
<td>
<p>The dataframe in which to search for overlapping SNPs from dataframe1</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>thresh</code></td>
<td>
<p>Numeric, the p-value threshold used for extracting the top snps from dataset 1</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>protein_coding_only</code></td>
<td>
<p>Logical, set this variable to TRUE to only use protein_coding genes for the annotation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>region_size</code></td>
<td>
<p>Integer, the size of the interval which to extract the top snps from</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Logical, (default: FALSE). Assign to TRUE to get information on which alleles are matched and which are not.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Dataframe containing the top hit
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
create_snpset(CD_UKBB,CD_FINNGEN, thresh=1e-09)

## End(Not run)

</code></pre>


</div>