<div class="container">

<table style="width: 100%;"><tr>
<td>annotate_with_nearest_gene</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get the nearest gene for one or more snps</h2>

<h3>Description</h3>

<p><code>annotate_with_nearest_gene()</code> Annotate the variant/snp with their nearest gene
Required parameters is a dataframe of SNPs (with the columns CHROM and POS)
</p>


<h3>Usage</h3>

<pre><code class="language-R">annotate_with_nearest_gene(
  variants,
  protein_coding_only = FALSE,
  build = 38,
  .chr_map = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>variants</code></td>
<td>
<p>a dataframe of variant positions (CHROM and POS)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>protein_coding_only</code></td>
<td>
<p>Logical, if set to TRUE only annotate with protein coding genes (the default value is FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>build</code></td>
<td>
<p>A number representing the genome build. Set to 37 to change to build (GRCh37). The default is build 38 (GRCh38).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.chr_map</code></td>
<td>
<p>An internally used list which maps chromosome names to numbers.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>the input dataframe with Gene_Symbol as an additional column
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
variants &lt;-get_lead_snps(CD_UKBB)
annotate_with_nearest_gene(variants)

## End(Not run)

</code></pre>


</div>