<div class="container">

<table style="width: 100%;"><tr>
<td>effectplot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create a plot comparing variant effects in two datasets</h2>

<h3>Description</h3>

<p><code>effectplot()</code>
</p>


<h3>Usage</h3>

<pre><code class="language-R">effectplot(
  df,
  pheno_x = "x_pheno",
  pheno_y = "y_pheno",
  annotate_with = "Gene_Symbol",
  thresh = 5e-08,
  ci_thresh = 1,
  gene_label_thresh = 5e-08,
  color = get_topr_colors()[1],
  scale = 1,
  build = 38,
  label_fontface = "italic",
  label_family = "",
  nudge_y = 0.001,
  nudge_x = 0.001,
  size = 2,
  segment.size = 0.2,
  segment.linetype = "solid",
  segment.color = "transparent",
  angle = 0,
  title = NULL,
  axis_text_size = 10,
  axis_title_size = 12,
  title_text_size = 13,
  subtitle_text_size = 11,
  gene_label_size = 3.2,
  snpset_thresh = 5e-08,
  snpset_region_size = 1e+06,
  max.overlaps = 10,
  annotate = 0,
  label_color = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>df</code></td>
<td>
<p>The input dataframe (snpset) containing one row per variant and P values (P1 and P2) and effects (E1 and E2) from two datasets/phenotypes OR a list containing two datasets.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno_x</code></td>
<td>
<p>A string representing the name of the phenotype whose effect is plotted on the x axis</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pheno_y</code></td>
<td>
<p>A string representing the name of the phenotype whose effect is plotted on the y axis</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annotate_with</code></td>
<td>
<p>A string, The name of the column that contains the label for the datapoints (default value is Gene_Symbol)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>thresh</code></td>
<td>
<p>A number. Threshold cutoff, datapoints with P2 below this threshold are shown as filled circles whereas datapoints with P2 above this threshold are shown as open circles</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ci_thresh</code></td>
<td>
<p>A number.Show the confidence intervals if the P-value is below this threshold</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene_label_thresh</code></td>
<td>
<p>Deprecated: A number, label datapoints with P2 below this threshold</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>color</code></td>
<td>
<p>A string, default value is the first of the topr colors</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scale</code></td>
<td>
<p>A number, to change the size of the title and axes labels and ticks at the same time (default : 1)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>build</code></td>
<td>
<p>A number representing the genome build or a data frame. Set to 37 to change to build (GRCh37). The default is build 38 (GRCh38).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label_fontface</code></td>
<td>
<p>A string or a vector of strings. Label font “plain”, “bold”, “italic”, “bold.italic” (ggrepel argument)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label_family</code></td>
<td>
<p>A string or a vector of strings. Label font name (default ggrepel argument is "")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nudge_y</code></td>
<td>
<p>A number to horizontally adjust the starting position of each gene label (this is a ggrepel parameter)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nudge_x</code></td>
<td>
<p>A number to vertically adjust the starting position of each gene label (this is a ggrepel parameter)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>size</code></td>
<td>
<p>A number or a vector of numbers, setting the size of the plot points (default: <code>size=1.2</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>segment.size</code></td>
<td>
<p>line segment color (ggrepel argument)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>segment.linetype</code></td>
<td>
<p>line segment solid, dashed, etc.(ggrepel argument)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>segment.color</code></td>
<td>
<p>line segment thickness (ggrepel argument)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>angle</code></td>
<td>
<p>A number, the angle of the text label</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title</code></td>
<td>
<p>A string to set the plot title</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>axis_text_size</code></td>
<td>
<p>A number, size of the x and y axes tick labels (default: 12)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>axis_title_size</code></td>
<td>
<p>A number, size of the x and y title labels (default: 12)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title_text_size</code></td>
<td>
<p>A number, size of the plot title (default: 13)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subtitle_text_size</code></td>
<td>
<p>A number setting the text size of the subtitle (default: 11)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene_label_size</code></td>
<td>
<p>A number setting the size of the gene labels shown at the bottom of the plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snpset_thresh</code></td>
<td>
<p>A number representing the threshold used to create the snpset used for plotting (Only applicable if the input dataframe is a list containing two datasets)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snpset_region_size</code></td>
<td>
<p>A number representing the region size to use when creating the snpset used for plotting (Only applicable if the input dataframe is a list containing two datasets)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.overlaps</code></td>
<td>
<p>Exclude text labels that overlap too many things. Defaults to 10 (ggrepel argument)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annotate</code></td>
<td>
<p>A number, label datapoints with p-value below below this number (in the second df) by their nearest gene</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label_color</code></td>
<td>
<p>A string or a vector of strings. To change the color of the gene or variant labels</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>ggplot object
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
effectplot(list(CD_UKBB, CD_FINNGEN))

## End(Not run)

</code></pre>


</div>