<div class="container">

<table style="width: 100%;"><tr>
<td>get_lead_snps</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get the index/lead variants</h2>

<h3>Description</h3>

<p><code>get_lead_snps()</code> Get the top variants within 1 MB windows of the genome with association p-values below the given threshold
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_lead_snps(
  df,
  thresh = 5e-08,
  region_size = 1e+06,
  protein_coding_only = FALSE,
  chr = NULL,
  .checked = FALSE,
  verbose = NULL,
  keep_chr = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>df</code></td>
<td>
<p>Dataframe</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>thresh</code></td>
<td>
<p>A number. P-value threshold, only extract variants with p-values below this threshold (5e-08 by default)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>region_size</code></td>
<td>
<p>An integer (default = 20000000) (or a string represented as 200kb or 2MB) indicating the window size for variant labeling. Increase this number for sparser annotation and decrease for denser annotation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>protein_coding_only</code></td>
<td>
<p>Logical, set this variable to TRUE to only use protein_coding genes for annotation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>String, get the top variants from one chromosome only, e.g. chr="chr1"</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.checked</code></td>
<td>
<p>Logical, if the input data has already been checked, this can be set to TRUE so it wont be checked again (FALSE by default)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Logical, set to TRUE to get printed information on number of SNPs extracted</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep_chr</code></td>
<td>
<p>Logical, set to FALSE to remove the "chr" prefix before each chromosome if present (TRUE by default)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Dataframe of lead variants. Returns the best variant per MB (by default, change the region size with the region argument) with p-values below the input threshold (thresh=5e-08 by default)
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
get_lead_snps(CD_UKBB)

## End(Not run)

</code></pre>


</div>