<div class="container">

<table style="width: 100%;"><tr>
<td>CD_UKBB</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>UKBB Crohns disease (ICD 10 code K50)</h2>

<h3>Description</h3>

<p>Dataset retrieved from the UK biobank consisting of 2,799 crohnÂ´s cases (K50) and 484,515 controls. The dataset has been filtered on variants with P &lt;1e-03.
</p>


<h3>Usage</h3>

<pre><code class="language-R">CD_UKBB
</code></pre>


<h3>Format</h3>

<p>A data frame  with 21,717 rows and 8 variables:
</p>

<dl>
<dt>CHROM</dt>
<dd>
<p>Chromosome, written as for example chr1 or 1</p>
</dd>
<dt>POS</dt>
<dd>
<p>genetic position of the variant</p>
</dd>
<dt>REF</dt>
<dd>
<p>the reference allele</p>
</dd>
<dt>ALT</dt>
<dd>
<p>the alternative allele</p>
</dd>
<dt>ID</dt>
<dd>
<p>Variant identifier, e.g. rsid </p>
</dd>
<dt>P</dt>
<dd>
<p>P-value from Plink run, additive model, regression model GLM_FIRTH</p>
</dd>
<dt>OR</dt>
<dd>
<p>Odds Ratio</p>
</dd>
<dt>AF</dt>
<dd>
<p>Allele frequency</p>
</dd>
</dl>
<h3>Source</h3>

<p>Crohn's UKBB ICD10 code K50, only including variants with P&lt;1e-03
</p>


</div>