<div class="container">

<table style="width: 100%;"><tr>
<td>get_genes_in_region</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get SNPs/variants within region</h2>

<h3>Description</h3>

<p><code>get_genes_in_region()</code>
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_genes_in_region(
  chr = chr,
  xmin = xmin,
  xmax = xmax,
  protein_coding_only = F,
  show_exons = F,
  show_genes = T,
  build = 38,
  region = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>A string, chromosome (e.g. chr16)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xmin</code></td>
<td>
<p>An integer representing genetic position</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xmax</code></td>
<td>
<p>An integer representing genetic position</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>protein_coding_only</code></td>
<td>
<p>A logical scalar, if TRUE, only protein coding genes are used for annotation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show_exons</code></td>
<td>
<p>Deprecated : A logical scalar, show exons instead of genes (default show_exons=FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show_genes</code></td>
<td>
<p>A logical scalar, show genes instead of exons (default show_genes=FALSE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>build</code></td>
<td>
<p>A number representing the genome build or a data frame. Set to 37 to change to build (GRCh37). The default is build 38 (GRCh38).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>region</code></td>
<td>
<p>A string representing the genetic region (e.g chr16:50693587-50734041)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>the genes the requested region
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
get_genes_in_region(region="chr16:50593587-50834041")

## End(Not run)
</code></pre>


</div>