<div class="container">

<table style="width: 100%;"><tr>
<td>manhattan</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create a Manhattan plot</h2>

<h3>Description</h3>

<p><code>manhattan()</code> displays association results for the entire genome on a Manhattan plot.
Required parameter is at least one dataset (dataframe) containing the association data (with columns <code>CHROM,POS,P</code> in upper or lowercase)
</p>
<p>All other input parameters are optional
</p>


<h3>Usage</h3>

<pre><code class="language-R">manhattan(
  df,
  ntop = 4,
  title = "",
  annotate = NULL,
  color = NULL,
  sign_thresh = 5e-08,
  sign_thresh_color = "red",
  sign_thresh_label_size = 3.5,
  label_size = 3.5,
  size = 0.8,
  shape = 19,
  alpha = 1,
  highlight_genes_color = "darkred",
  highlight_genes_ypos = 1.5,
  axis_text_size = 12,
  axis_title_size = 14,
  title_text_size = 15,
  legend_title_size = 13,
  legend_text_size = 12,
  protein_coding_only = TRUE,
  angle = 0,
  legend_labels = NULL,
  chr = NULL,
  annotate_with = "Gene_Symbol",
  region_size = 2e+07,
  legend_name = NULL,
  legend_position = "bottom",
  nudge_x = 0.1,
  nudge_y = 0.7,
  xmin = NULL,
  xmax = NULL,
  ymin = NULL,
  ymax = NULL,
  highlight_genes = NULL,
  label_color = NULL,
  legend_nrow = NULL,
  gene_label_size = NULL,
  gene_label_angle = 0,
  scale = 1,
  show_legend = TRUE,
  sign_thresh_linetype = "dashed",
  sign_thresh_size = 0.5,
  rsids = NULL,
  rsids_color = NULL,
  rsids_with_vline = NULL,
  annotate_with_vline = NULL,
  shades_color = NULL,
  shades_alpha = 0.5,
  segment.size = 0.2,
  segment.color = "black",
  segment.linetype = "dashed",
  max.overlaps = 10,
  label_fontface = "plain",
  label_family = "",
  gene_label_fontface = "plain",
  gene_label_family = "",
  build = 38,
  verbose = NULL,
  label_alpha = 1,
  shades_line_alpha = 1,
  vline = NULL,
  vline_color = "grey",
  vline_linetype = "dashed",
  vline_alpha = 1,
  vline_size = 0.5,
  region = NULL,
  theme_grey = FALSE,
  xaxis_label = "Chromosome",
  use_shades = FALSE,
  even_no_chr_lightness = 0.8,
  get_chr_lengths_from_data = TRUE,
  log_trans_p = TRUE,
  chr_ticknames = NULL,
  show_all_chrticks = FALSE,
  hide_chrticks_from_pos = 17,
  hide_chrticks_to_pos = NULL,
  hide_every_nth_chrtick = 2,
  downsample_cutoff = 0.05,
  downsample_prop = 0.1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>df</code></td>
<td>
<p>Dataframe or a list of dataframes (required columns are <code>CHROM,POS,P</code>), in upper- or lowercase) of association results.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ntop</code></td>
<td>
<p>An integer, number of datasets (GWAS results) to show on the top plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title</code></td>
<td>
<p>A string to set the plot title</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annotate</code></td>
<td>
<p>A number (p-value). Display annotation for variants with p-values below this threshold</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>color</code></td>
<td>
<p>A string or a vector of strings, for setting the color of the datapoints on the plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sign_thresh</code></td>
<td>
<p>A number or vector of numbers, setting the horizontal significance threshold (default: <code>sign_thresh=5e-8</code>). Set to NULL to hide the significance threshold.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sign_thresh_color</code></td>
<td>
<p>A string or vector of strings to set the color/s of the significance threshold/s</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sign_thresh_label_size</code></td>
<td>
<p>A number setting the text size of the label for the significance thresholds (default text size is 3.5)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label_size</code></td>
<td>
<p>An number to set the size of the plot labels (default: <code>label_size=3</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>size</code></td>
<td>
<p>A number or a vector of numbers, setting the size of the plot points (default: <code>size=1.2</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>shape</code></td>
<td>
<p>A number of a vector of numbers setting the shape of the plotted points</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>A number or a vector of numbers setting the transparency of the plotted points</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>highlight_genes_color</code></td>
<td>
<p>A string, color for the highlighted genes (default: darkred)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>highlight_genes_ypos</code></td>
<td>
<p>An integer, controlling where on the y-axis the highlighted genes are placed (default value is 1)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>axis_text_size</code></td>
<td>
<p>A number, size of the x and y axes tick labels (default: 12)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>axis_title_size</code></td>
<td>
<p>A number, size of the x and y title labels (default: 12)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title_text_size</code></td>
<td>
<p>A number, size of the plot title (default: 13)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend_title_size</code></td>
<td>
<p>A number, size of the legend title</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend_text_size</code></td>
<td>
<p>A number, size of the legend text</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>protein_coding_only</code></td>
<td>
<p>A logical scalar, if TRUE, only protein coding genes are used for annotation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>angle</code></td>
<td>
<p>A number, the angle of the text label</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend_labels</code></td>
<td>
<p>A string or vector of strings representing legend labels for the input datasets</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>A string or integer, the chromosome to plot (i.e. chr15), only required if the input dataframe contains results from more than one chromosome</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annotate_with</code></td>
<td>
<p>A string. Annotate the variants with either Gene_Symbol or ID (default: "Gene_Symbol")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>region_size</code></td>
<td>
<p>An integer (default = 20000000) (or a string represented as 200kb or 2MB) indicating the window size for variant labeling. Increase this number for sparser annotation and decrease for denser annotation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend_name</code></td>
<td>
<p>A string, use to change the name of the legend (default: None)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend_position</code></td>
<td>
<p>A string, top,bottom,left or right</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nudge_x</code></td>
<td>
<p>A number to vertically adjust the starting position of each gene label (this is a ggrepel parameter)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nudge_y</code></td>
<td>
<p>A number to horizontally adjust the starting position of each gene label (this is a ggrepel parameter)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xmin, xmax</code></td>
<td>
<p>Integer, setting the chromosomal range to display on the x-axis</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ymin, ymax</code></td>
<td>
<p>Integer, min and max of the y-axis, (default values: <code>ymin=0, ymax=max(-log10(df$P))</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>highlight_genes</code></td>
<td>
<p>A string or vector of strings, gene or genes to highlight at the bottom of the plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label_color</code></td>
<td>
<p>A string or a vector of strings. To change the color of the gene or variant labels</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend_nrow</code></td>
<td>
<p>An integer, sets the number of rows allowed for the legend labels</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene_label_size</code></td>
<td>
<p>A number setting the size of the gene labels shown at the bottom of the plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene_label_angle</code></td>
<td>
<p>A number setting the angle of the gene label shown at the bottom of the plot (default: 0)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scale</code></td>
<td>
<p>A number, to change the size of the title and axes labels and ticks at the same time (default : 1)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show_legend</code></td>
<td>
<p>A logical scalar, set to FALSE to hide the legend (default : TRUE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sign_thresh_linetype</code></td>
<td>
<p>A string, the line-type of the horizontal significance threshold (default : dashed)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sign_thresh_size</code></td>
<td>
<p>A number, sets the size of the horizontal significance threshold line (default : 1)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rsids</code></td>
<td>
<p>A string (rsid) or vector of strings to highlight on the plot, e.g. <code>rsids=c("rs1234, rs45898")</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rsids_color</code></td>
<td>
<p>A string, the color of the variants in variants_id (default color is red)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rsids_with_vline</code></td>
<td>
<p>A string (rsid) or vector of strings to highlight on the plot with their rsids and vertical lines further highlighting their positions</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annotate_with_vline</code></td>
<td>
<p>A number (p-value). Display annotation and vertical lines for variants with p-values below this threshold</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>shades_color</code></td>
<td>
<p>The color of the rectangles (shades) representing the different chromosomes on the Manhattan plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>shades_alpha</code></td>
<td>
<p>The transparency (alpha) of the rectangles (shades)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>segment.size</code></td>
<td>
<p>line segment color (ggrepel argument)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>segment.color</code></td>
<td>
<p>line segment thickness (ggrepel argument)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>segment.linetype</code></td>
<td>
<p>line segment solid, dashed, etc.(ggrepel argument)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>max.overlaps</code></td>
<td>
<p>Exclude text labels that overlap too many things. Defaults to 10 (ggrepel argument)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label_fontface</code></td>
<td>
<p>A string or a vector of strings. Label font “plain”, “bold”, “italic”, “bold.italic” (ggrepel argument)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label_family</code></td>
<td>
<p>A string or a vector of strings. Label font name (default ggrepel argument is "")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene_label_fontface</code></td>
<td>
<p>Gene label font “plain”, “bold”, “italic”, “bold.italic” (ggrepel argument)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene_label_family</code></td>
<td>
<p>Gene label font name (default ggrepel argument is "")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>build</code></td>
<td>
<p>A number representing the genome build or a data frame. Set to 37 to change to build (GRCh37). The default is build 38 (GRCh38).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>A logical scalar (default: NULL). Set to FALSE to suppress printed messages</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label_alpha</code></td>
<td>
<p>An number or vector of numbers to set the transparency of the plot labels (default: <code>label_alpha=1</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>shades_line_alpha</code></td>
<td>
<p>The transparency (alpha) of the lines around the rectangles (shades)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vline</code></td>
<td>
<p>A number or vector of numbers to add a vertical line to the plot at a specific chromosomal position, e.g <code>vline="chr1:204000066"</code>. Multiple values can be provided in a vector, e.g  <code>vline=c("chr1:204000066","chr5:100500188")</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vline_color</code></td>
<td>
<p>A string. The color of added vertical line/s (default: grey)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vline_linetype</code></td>
<td>
<p>A string. The linetype of added vertical line/s (default : dashed)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vline_alpha</code></td>
<td>
<p>A number. The alpha of added vertical line/s (default : 1)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vline_size</code></td>
<td>
<p>A number.The size of added vertical line/s (default : 0.5)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>region</code></td>
<td>
<p>A string representing a genetic region, e.g. chr1:67038906-67359979</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>theme_grey</code></td>
<td>
<p>A logical scalar (default: FALSE). Use gray rectangles (instead of white to distinguish between chromosomes)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xaxis_label</code></td>
<td>
<p>A string. The label for the x-axis (default: Chromosome)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use_shades</code></td>
<td>
<p>A logical scalar (default: FALSE). Use shades/rectangles to distinguish between chromosomes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>even_no_chr_lightness</code></td>
<td>
<p>Lightness value for even numbered chromosomes. A number or vector of numbers between 0 and 1 (default: 0.8). If set to 0.5, the same color as shown for odd numbered chromosomes is displayed. A value below 0.5 will result in a darker color displayed for even numbered chromosomes, whereas a value above 0.5 results in a lighter color.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>get_chr_lengths_from_data</code></td>
<td>
<p>A logical scalar (default: TRUE). If set to FALSE, use the inbuilt chromosome lengths (from hg38), instead of chromosome lengths based on the max position for each chromosome in the input dataset/s.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log_trans_p</code></td>
<td>
<p>A logical scalar (default: TRUE). By default the p-values in the input datasets are log transformed using -log10. Set this argument to FALSE if the p-values in the datasets have already been log transformed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr_ticknames</code></td>
<td>
<p>A vector containing the chromosome names displayed on the x-axis. If NULL, the following format is used: chr_ticknames &lt;- c(1:16, ”,18, ”,20, ”,22, 'X')</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show_all_chrticks</code></td>
<td>
<p>A logical scalar (default : FALSE). Set to TRUE to show all the chromosome names on the ticks on the x-axis</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hide_chrticks_from_pos</code></td>
<td>
<p>A number (default: 17). Hide every nth chromosome name on the x-axis FROM this position (chromosome number)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hide_chrticks_to_pos</code></td>
<td>
<p>A number (default: NULL). Hide every nth chromosome name on the x-axis TO this position (chromosome number). When NULL this variable will be set to the number of numeric chromosomes in the input dataset.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hide_every_nth_chrtick</code></td>
<td>
<p>A number (default: 2). Hide every nth chromosome tick on the x-axis (from the hide_chr_ticks_from_pos to the hide_chr_ticks_to_pos).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>downsample_cutoff</code></td>
<td>
<p>A number (default: 0.05) used to downsample the input dataset prior to plotting. Sets the fraction of high p-value (default: P&gt;0.05) markers to display on the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>downsample_prop</code></td>
<td>
<p>A number (default: 0.1) used to downsample the input dataset prior to plotting. Only a proportion of the variants (10% by default) with P-values higher than the downsample_cutoff will be displayed on the plot.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>ggplot object
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
manhattan(CD_UKBB)

## End(Not run)
</code></pre>


</div>