<div class="container">

<table style="width: 100%;"><tr>
<td>get_sign_and_sugg_loci</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get the index/lead variants</h2>

<h3>Description</h3>

<p><code>get_lead_snps()</code> Get the top variants within 1 MB windows of the genome with association p-values below the given threshold
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_sign_and_sugg_loci(
  df,
  genome_wide_thresh = 5e-08,
  suggestive_thresh = 1e-06,
  flank_size = 1e+06,
  region_size = 1e+06
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>df</code></td>
<td>
<p>Dataframe, GWAS summary statistics</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome_wide_thresh</code></td>
<td>
<p>A number. P-value threshold for genome wide significant loci (5e-08 by default)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>suggestive_thresh</code></td>
<td>
<p>A number. P-value threshold for suggestive loci (1e-06 by default)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>flank_size</code></td>
<td>
<p>A number (default = 1e6). The size of the flanking region for the significant and suggestitve snps.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>region_size</code></td>
<td>
<p>A number (default = 1e6). The size of the region for top snp search. Only one snp per region is returned.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>List of genome wide and suggestive loci.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
get_sign_and_sugg_loci(CD_UKBB)

## End(Not run)

</code></pre>


</div>