<div class="container">

<table style="width: 100%;"><tr>
<td>manhattanExtra</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create a Manhattan plot highlighting genome-wide significant and suggestive loci</h2>

<h3>Description</h3>

<p><code>manhattanExtra()</code> displays association results for the entire genome on a Manhattan plot, highlighting genome-wide significant and suggestive loci.
Required parameter is at least one dataset (dataframe) containing the association data (with columns <code>CHROM,POS,P</code> in upper or lowercase)
</p>
<p>All other input parameters are optional
</p>


<h3>Usage</h3>

<pre><code class="language-R">manhattanExtra(
  df,
  genome_wide_thresh = 5e-08,
  suggestive_thresh = 1e-06,
  flank_size = 1e+06,
  region_size = 1e+06,
  sign_thresh_color = NULL,
  sign_thresh_label_size = NULL,
  show_legend = TRUE,
  label_fontface = NULL,
  nudge_y = NULL,
  ymax = NULL,
  sign_thresh = NULL,
  label_color = NULL,
  color = NULL,
  legend_labels = NULL,
  annotate = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>df</code></td>
<td>
<p>Dataframe, GWAS summary statistics</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genome_wide_thresh</code></td>
<td>
<p>A number. P-value threshold for genome wide significant loci (5e-08 by default)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>suggestive_thresh</code></td>
<td>
<p>A number. P-value threshold for suggestive loci (1e-06 by default)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>flank_size</code></td>
<td>
<p>A number (default = 1e6). The size of the flanking region for the significant and suggestitve snps.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>region_size</code></td>
<td>
<p>A number (default = 1e6). The size of the region for gene annotation. Increase this number for sparser annotation and decrease for denser annotation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sign_thresh_color</code></td>
<td>
<p>A string or vector of strings to set the color/s of the significance threshold/s</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sign_thresh_label_size</code></td>
<td>
<p>A number setting the text size of the label for the significance thresholds (default text size is 3.5)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show_legend</code></td>
<td>
<p>A logical scalar, set to FALSE to hide the legend (default : TRUE)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label_fontface</code></td>
<td>
<p>A string or a vector of strings. Label font “plain”, “bold”, “italic”, “bold.italic” (ggrepel argument)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nudge_y</code></td>
<td>
<p>A number to horizontally adjust the starting position of each gene label (this is a ggrepel parameter)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ymax</code></td>
<td>
<p>Integer, max of the y-axis, (default value: <code>ymax=(max(-log10(df$P)) + max(-log10(df$P)) * .2))</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sign_thresh</code></td>
<td>
<p>A number or vector of numbers, setting the horizontal significance threshold (default: <code>sign_thresh=5e-8</code>). Set to NULL to hide the significance threshold.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>label_color</code></td>
<td>
<p>A string or a vector of strings. To change the color of the gene or variant labels</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>color</code></td>
<td>
<p>A string or a vector of strings, for setting the color of the datapoints on the plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend_labels</code></td>
<td>
<p>A string or vector of strings representing legend labels for the input datasets</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annotate</code></td>
<td>
<p>A number (p-value). Display annotation for variants with p-values below this threshold</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional arguments passed to other plotting functions.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>ggplot object
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
manhattanExtra(df)

## End(Not run)

</code></pre>


</div>