<div class="container">

<table style="width: 100%;"><tr>
<td>get_snps_within_region</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get SNPs/variants within region</h2>

<h3>Description</h3>

<p><code>get_snps_within_region()</code>
</p>


<h3>Usage</h3>

<pre><code class="language-R">get_snps_within_region(
  df,
  region,
  chr = NULL,
  xmin = NULL,
  xmax = NULL,
  keep_chr = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>df</code></td>
<td>
<p>data frame of association results with the columns CHR and POS</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>region</code></td>
<td>
<p>A string representing the genetic region (e.g chr16:50693587-50734041)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>
<p>A string, chromosome (e.g. chr16)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xmin</code></td>
<td>
<p>An integer, include variants with POS larger than xmin</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xmax</code></td>
<td>
<p>An integer, include variants with POS smaller than xmax</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep_chr</code></td>
<td>
<p>Deprecated: Logical, set to FALSE to remove the "chr" prefix before each chromosome if present (TRUE by default)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>the variants within the requested region
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
get_snps_within_region(CD_UKBB, "chr16:50593587-50834041")

## End(Not run)

</code></pre>


</div>