<div class="container">

<table style="width: 100%;"><tr>
<td>ncbi_byid</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Retrieve gene sequences from NCBI by accession number.</h2>

<h3>Description</h3>

<p>Retrieve gene sequences from NCBI by accession number.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ncbi_byid(ids, format = NULL, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ids</code></td>
<td>
<p>(character) GenBank ids to search for. One or more. Required.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>format</code></td>
<td>
<p>(character) Return type, e.g., <code>"fasta"</code>. NOW IGNORED.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>(logical) If <code>TRUE</code> (default), informative messages
printed.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If bad ids are included with good ones, the bad ones are
silently dropped. If all ids are bad you'll get a stop with error message.
</p>


<h3>Value</h3>

<p>data.frame of the form:
</p>

<ul>
<li>
<p> taxon - taxonomic name (may include some junk, but hard to parse off)
</p>
</li>
<li>
<p> taxonomy - organism lineage
</p>
</li>
<li>
<p> gene_desc - gene description
</p>
</li>
<li>
<p> organelle - if mitochondrial or chloroplast
</p>
</li>
<li>
<p> gi_no - GI number
</p>
</li>
<li>
<p> acc_no - accession number
</p>
</li>
<li>
<p> keyword - if official DNA barcode
</p>
</li>
<li>
<p> specimen_voucher - museum/lab accession number of vouchered material
</p>
</li>
<li>
<p> lat_lon - longitude/latitude of specimen collection event
</p>
</li>
<li>
<p> country - country/location of specimen collection event
</p>
</li>
<li>
<p> paper_title - title of study
</p>
</li>
<li>
<p> journal - journal study published in (if published)
</p>
</li>
<li>
<p> first_author - first author of study
</p>
</li>
<li>
<p> uploaded_date - date sequence was uploaded to GenBank
</p>
</li>
<li>
<p> length - sequence length
</p>
</li>
<li>
<p> sequence - sequence character string
</p>
</li>
</ul>
<h3>Author(s)</h3>

<p>Scott Chamberlain, Rupert Collins
</p>


<h3>See Also</h3>

<p><code>ncbi_searcher()</code>, ncbi_byname()]
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# A single gene
ncbi_byid(ids="360040093")

# Many genes (with different accession numbers)
ncbi_byid(ids=c("360040093","347448433"))

## End(Not run)
</code></pre>


</div>