<div class="container">

<table style="width: 100%;"><tr>
<td>trancriptVis</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>trancriptVis</h2>

<h3>Description</h3>

<p>This package is to visualize gene diffrent isoforms.
</p>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gtfFile</code></td>
<td>
<p>GTF file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene</code></td>
<td>
<p>Target gene to plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>myTranscript</code></td>
<td>
<p>Specify which transcripts to plot use transcipt id.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Chr</code></td>
<td>
<p>Chromosome number.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>posStart</code></td>
<td>
<p>Region start position on genome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>posEnd</code></td>
<td>
<p>Region end position on genome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>collapse</code></td>
<td>
<p>Whether to collapse multiple transcripts into one, default(FALSE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exonWidth</code></td>
<td>
<p>Exon width to plot, default(0.3).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>relTextDist</code></td>
<td>
<p>Transcripts name or gene name relative to exon, default(0.3).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>intronSize</code></td>
<td>
<p>Intron line size, default(0.5).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>arrowBreak</code></td>
<td>
<p>How many gap distance to draw arrows, the smaller the more arrows, default(0.15).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exonColorBy</code></td>
<td>
<p>Whether color group by "transcript_id" or "gene_name", default(NULL).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exonFill</code></td>
<td>
<p>Exon fill color, default('#333399').</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>circle</code></td>
<td>
<p>Whether make plot into a circle plot, default(FALSE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cicStart</code></td>
<td>
<p>Circle plot start position, default(pi).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>circSegCol</code></td>
<td>
<p>Circle sgement color, default('#333399').</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>text_only</code></td>
<td>
<p>When circle plot labeled by gene name, whether remove the line connected with gene name, default(FALSE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylimLow</code></td>
<td>
<p>The Y axis lower limitation of Circle plot, default(-10).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>openAngle</code></td>
<td>
<p>The gap of the circle plot, default(0.5).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>arrowCol</code></td>
<td>
<p>Normal arrow color, default('#333399').</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>arrowAngle</code></td>
<td>
<p>Normal arrow angle, default(30).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>arrowLength</code></td>
<td>
<p>Normal arrow length, default(0.1).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>arrowType</code></td>
<td>
<p>Normal arrow type, default('open').</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>addNormalArrow</code></td>
<td>
<p>Whether add normal arrow on plot, default(TRUE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>newStyleArrow</code></td>
<td>
<p>Whether add new style arrow on plot, default(FALSE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>absSpecArrowLen</code></td>
<td>
<p>Whether make new style arrow length to be relative to each transcript length or absolute length to the longest transcript, default(FALSE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>speArrowRelPos</code></td>
<td>
<p>The relative position to the transcript on horizontal direction of new style arrow, default(0).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>speArrowRelLen</code></td>
<td>
<p>The relative length to the transcript length of new style arrow, default(0.05).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>speArrowStart</code></td>
<td>
<p>The new style arrow start position on the vertical direction, default(-0.15).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>speArrowRelHigh</code></td>
<td>
<p>The relative height of new style arrow to the vertical length, default(2).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>speArrowLineSize</code></td>
<td>
<p>The new style arrow line size, default(0.5).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>speArrowCol</code></td>
<td>
<p>The new style arrow line color, default('black').</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>speArrowAngle</code></td>
<td>
<p>The new style arrow angle, default(30).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>speArrowLen</code></td>
<td>
<p>The new style arrow length, default(0.1).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>speArrowType</code></td>
<td>
<p>The new style arrow type, default('closed').</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>textLabel</code></td>
<td>
<p>The text label aesthetic mappings, default('transcript_id').</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>textLabelSize</code></td>
<td>
<p>The text label size, default(5).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>textLabelColor</code></td>
<td>
<p>The text label color, default('black').</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>base_size</code></td>
<td>
<p>Theme basesize, default(14).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>marginX</code></td>
<td>
<p>Plot left and right margins, default(0.2).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>marginY</code></td>
<td>
<p>Plot top and bottomn margins, default(0.2).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aspect.ratio</code></td>
<td>
<p>Plot ratio, default(NULL).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>facetByGene</code></td>
<td>
<p>Whether facet by gene to plot, this useful for your genes which are far away from each other or not located on the same chromosome, default(FALSE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncolGene</code></td>
<td>
<p>The column numbers to plot, default(NULL).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scales</code></td>
<td>
<p>Facet plot scales, same as "facet_wrap" function, default('free').</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>strip.position</code></td>
<td>
<p>Facet plot strip.position, same as "facet_wrap" function, default('top').</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>forcePosRel</code></td>
<td>
<p>Whether force the genome coordinate to relative position to transcript start/end position, default('FALSE').</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>panel.spacing</code></td>
<td>
<p>Facet plot panel space, default(0.3).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>revNegStrand</code></td>
<td>
<p>Whether reverse the negtive strand when set "forcePosRel=TRUE", default('FALSE').</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A ggplot object.
</p>


<h3>Author(s)</h3>

<p>Junjun Lao
</p>


<h3>Examples</h3>

<pre><code class="language-R">##############################################################
# test function

########################################################
# load data
data(gtf)

# non-coding gene
trancriptVis(gtfFile = gtf,
             gene = 'Xist')

# coding gene
trancriptVis(gtfFile = gtf,
             gene = 'Nanog')

# change fill color
trancriptVis(gtfFile = gtf,
             gene = 'Nanog',
             exonFill = '#CCFF00')

# change inrton line size
trancriptVis(gtfFile = gtf,
             gene = 'Nanog',
             intronSize = 1)

# change label size,color and position
trancriptVis(gtfFile = gtf,
             gene = 'Nanog',
             textLabelSize = 4,
             textLabelColor = 'red',
             relTextDist = 0)

# aes by gene name
trancriptVis(gtfFile = gtf,
             gene = 'Nanog',
             textLabel = 'gene_name')

# color aes by transcript
trancriptVis(gtfFile = gtf,
             gene = 'Tpx2',
             exonColorBy = 'transcript_id')

# change arrow color and type
trancriptVis(gtfFile = gtf,
             gene = 'Nanog',
             arrowCol = 'orange',
             arrowType = 'closed')

# no intron gene and add arrow color
# change arrow color and type
trancriptVis(gtfFile = gtf,
             gene = 'Jun',
             textLabel = 'gene_name',
             arrowCol = 'white',
             arrowType = 'closed') +
  theme_void()

# add arrow breaks
trancriptVis(gtfFile = gtf,
             gene = 'Nanog',
             arrowCol = 'orange',
             arrowType = 'closed',
             arrowBreak = 0.1)

# draw specific transcript
p1 &lt;- trancriptVis(gtfFile = gtf,
                   gene = 'Commd7')

p2 &lt;- trancriptVis(gtfFile = gtf,
                   gene = 'Commd7',
                   myTranscript = c('ENSMUST00000071852','ENSMUST00000109782'))

# combine
cowplot::plot_grid(p1,p2,ncol = 2,align = 'hv')
</code></pre>


</div>