<div class="container">

<table style="width: 100%;"><tr>
<td>ll</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Fit Loglinear model with and without adjustment for TRD.
</h2>

<h3>Description</h3>

<p>Use input data, either sample of case-trios ('sample') or count data for 15/16-MFC-genotype categories ('counts') to fit loglinear model with (Model 2) and without (Model 1) adjustment  for TRD, model type could be genotype relative risk ('grr'), dominant ('dom'), gene-dosage ('gd'), genotype relative risk with imprinting variable ('grri'), dominant with imprinting variable ('domi'), or gene-dosage with imprinting variable ('gdi'). Note that the models fit only complete trios with no missing data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ll(data, data.type, tmother, tfather, model.type)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>

<p>input data can be
(1) case- and/or control-trios dataset with first column as disease status (1=case, 0=control), the second, third and forth columns as maternal, paternal and child genotypes. The fifth column, if entered as part of the input data, is the parent-of-origin indicator of disease allele in a heterozygous child (a value of 1 indicates the mother transmitted the disease allele and 0 indicates the father transmitted). The parent-of-origin indicator is only meaningful for a heterozygous child.
(2) a count dataset for the 15 or 16-MFC-genotype categories obtainable from the function strata.cnt(..), with the latter having 2 categories (111M &amp; 111F) for triply heterozygous trios.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data.type</code></td>
<td>

<p>type of data: 'sample' or 'counts'
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tmother</code></td>
<td>

<p>transmission ratio of minor alleles from mother to child
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tfather</code></td>
<td>

<p>transmission ratio of minor alleles from father to child
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>model.type</code></td>
<td>

<p>'grr', 'dom', 'gd', 'grri', 'domi', or 'gdi'
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>If the following warning messages occur, increase the population and sample size.
Warning messages:
glm.fit: fitted rates numerically 0 occurred
</p>


<h3>Value</h3>

<table><tr style="vertical-align: top;">
<td><code>Summary table</code></td>
<td>
<p>Summary statistics of fitting the loglinear model</p>
</td>
</tr></table>
<h3>Author(s)</h3>

<p>Lam Opal Huang
</p>


<h3>References</h3>

<p>Huang LO, Infante-Rivard C, Labbe A, (2015). Modeling sex-of-parent-specific Transmission Ratio Distortion and imprinting effect in loglinear model using case-trios.(Working manuscript)
</p>
<p>Huang LO, Infante-Rivard C, Labbe A, (2015). Adjustment for Transmission Ratio Distortion in the analysis of case-parent trios using a loglinear model.(Working manuscript)
</p>
<p>Weinberg CR, Wilcox AJ, Lie RT (1998) A log-linear approach to case-parent-triad data: assessing effects of disease genes that act either directly or through maternal effects and that may be subject to parental imprinting. American Journal of Human Genetics 62: 969-78.
</p>


<h3>See Also</h3>

<p><code>rtrios</code>
<code>strata.cnt</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">trios=rtrios(100000,500,0.1,0.2,0.3,1,0.5,0.5,0.1)
out=ll(trios$case,'sample',0.5,0.5,'grr')

cnt=strata.cnt(trios$case)
out=ll(cnt,'counts',0.5,0.5,'grr')

</code></pre>


</div>