<div class="container">

<table style="width: 100%;"><tr>
<td>treedater-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>treedater fits a molecular clock to a phylogenetic tree and estimates evolutionary rates and times of common ancestry.</h2>

<h3>Description</h3>

<p>Additional functions are provided for detecting outlier lineages (possible sequencing or alignment error). A statistical test is available for choosing between strict and relaxed clock models. The calendar time of each sample must be specified (possibly with bounds of uncertainty) and the length of the sequences used to estimate the tree. treedater uses heuristic search to optimise the TMRCAs of a phylogeny and the substitution rate. An uncorrelated relaxed molecular clock accounts for rate variation between lineages of the phylogeny which is parameterised using a Gamma-Poisson mixture model.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Erik Volz <a href="mailto:erik.volz@gmail.com">erik.volz@gmail.com</a>
</p>


<h3>References</h3>

<p>Volz, E. M., and S. D. W. Frost. "Scalable relaxed clock phylogenetic dating." Virus Evolution 3.2 (2017).
</p>


</div>