<div class="container">

<table style="width: 100%;"><tr>
<td>MappingQuality</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Faithfulness of mapped distances</h2>

<h3>Description</h3>

<p><code>MappingQuality()</code> calculates the trustworthiness and continuity
of mapped distances (Venna and Kaski 2001; Kaski et al. 2003).
Trustworthiness measures, on a scale from 0–1,
the degree to which points that are nearby in a mapping are truly close
neighbours; continuity, the extent to which points that are truly nearby
retain their close spatial proximity in a mapping.
</p>


<h3>Usage</h3>

<pre><code class="language-R">MappingQuality(original, mapped, neighbours = 10L)

ProjectionQuality(original, mapped, neighbours = 10L)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>original, mapped</code></td>
<td>
<p>Square matrix or <code>dist</code> object containing
original / mapped pairwise distances.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>neighbours</code></td>
<td>
<p>Integer specifying number of nearest neighbours to use in
calculation.  This should typically be small relative to the number of
points.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p><code>MappingQuality()</code> returns a named vector of length four,
containing the entries: <code>Trustworthiness</code>, <code>Continuity</code>, <code>TxC</code>
(the product of these values), and <code>sqrtTxC</code> (its square root).
</p>


<h3>Author(s)</h3>

<p><a href="https://orcid.org/0000-0001-5660-1727">Martin R. Smith</a>
(<a href="mailto:martin.smith@durham.ac.uk">martin.smith@durham.ac.uk</a>)
</p>


<h3>References</h3>

<p>Kaski S, Nikkila J, Oja M, Venna J, Toronen P, Castren E (2003).
“Trustworthiness and metrics in visualizing similarity of gene expression.”
<em>BMC Bioinformatics</em>, <b>4</b>, 48.
<a href="https://doi.org/10.1186/1471-2105-4-48">doi:10.1186/1471-2105-4-48</a>.<br><br> Venna J, Kaski S (2001).
“Neighborhood preservation in nonlinear projection methods: an experimental study.”
In Dorffner G, Bischof H, Hornik K (eds.), <em>Artificial Neural Networks — ICANN 2001</em>,  Lecture Notes in Computer Science, 485–491.
<a href="https://doi.org/10.1007/3-540-44668-0_68">doi:10.1007/3-540-44668-0_68</a>.
</p>


<h3>See Also</h3>

<p>Other tree space functions: 
<code>Islands()</code>,
<code>MSTSegments()</code>,
<code>MapTrees()</code>,
<code>SpectralEigens()</code>,
<code>cluster-statistics</code>,
<code>median.multiPhylo()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">library("TreeTools", quietly = TRUE)
trees &lt;- as.phylo(0:10, nTip = 8)
distances &lt;- ClusteringInfoDistance(trees)
mapping &lt;- cmdscale(distances)
MappingQuality(distances, dist(mapping), 4)
</code></pre>


</div>