<div class="container">

<table style="width: 100%;"><tr>
<td>plot_trumpets</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Trumpets</h2>

<h3>Description</h3>

<p>This function generates trumpet plots
</p>


<h3>Usage</h3>

<pre><code class="language-R">plot_trumpets(
  dataset = toy_data,
  rsID = "rsID",
  freq = "freq",
  A1_beta = "A1_beta",
  Analysis = "Analysis",
  Gene = "Gene",
  calculate_power = TRUE,
  show_power_curves = TRUE,
  exist_datapwr = NULL,
  threshold = c(0.7, 0.9),
  N = 1e+05,
  alpha = 5e-08,
  Nfreq = 500,
  power_color_palette = c("purple", "deeppink"),
  analysis_color_palette = c("#018571", "#a6611a")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dataset</code></td>
<td>
<p>Input text file with genetic association results. Columns required are rsID, freq, A1_beta, Analysis and Gene.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rsID</code></td>
<td>
<p>(required) Single Nucleotide Polymorphism (SNP) name.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>freq</code></td>
<td>
<p>(required) allele frequency of effect SNP.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>A1_beta</code></td>
<td>
<p>(required) risk allele effect size.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Analysis</code></td>
<td>
<p>(optional) adds colour to the type of analysis (e.g. GWAS, Sequencing).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Gene</code></td>
<td>
<p>(optional) Candidate gene name (can be empty).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>calculate_power</code></td>
<td>
<p>(TRUE/FALSE) Calculate power curves. Choose TRUE to add power curves for a given threshold, alpha, sample size N and number of allele frequencies. Choose FALSE if you already ran powerCurves() outside or do not want to show power curves.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>show_power_curves</code></td>
<td>
<p>(TRUE/FALSE) Show power curves in plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exist_datapwr</code></td>
<td>
<p>Existing dataframe containing columns: freq, pos.b.for.f, neg.b.for.f, powerline.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>
<p>Required if power == TRUE. Can be a single number or a vector of statistical power thresholds.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>N</code></td>
<td>
<p>(Required if calculate_power == TRUE). Sample size used to test the association.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>(Required if calculate_power == TRUE).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Nfreq</code></td>
<td>
<p>(Required if calculate_power == TRUE). Number of allele frequency data points generated to calculate the power curves. We recommend Nfreq&gt;1000 for power curves with high resolution. Note that this will slow down the rendering of the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>power_color_palette</code></td>
<td>
<p>A vector of colours for the power curves. Number of colors should match number of thresholds supplied.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>analysis_color_palette</code></td>
<td>
<p>A vector of colours for the analysis types.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Creates a Trumpet plot with variant allele frequency (X axis, log10 scale) and effect size information (Y axis).
</p>


<h3>Examples</h3>

<pre><code class="language-R">plot_trumpets(dataset = toy_data)
</code></pre>


</div>