<div class="container">

<table style="width: 100%;"><tr>
<td>featureSelect</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Trait specific gene selection using SVM and GA
</h2>

<h3>Description</h3>

<p>This function gives the optimal set of informative genes based on RNA-Seq count data
</p>


<h3>Usage</h3>

<pre><code class="language-R">featureSelect(X, y, p = 5, n.iter = 1, alpha = 0.05, p.adj.method = "bonferroni")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>X is a G x N data frame of gene expression values (raw count data) where rows represent genes and columns represent samples. Each cell entry represents the read counts of of a gene in a sample (row names of X as gene names or gene ids)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>y is a N x 1 numeric vector with entries 0 or 1 representing sample labels, where, 0/1 represents the sample label of samples for two conditions, e.g., 0 for Control and 1 for Case</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p</code></td>
<td>
<p>Population size, by default 5</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.iter</code></td>
<td>
<p>The number of iterations, by default 1</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>The level of significance, by default 0.05</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.adj.method</code></td>
<td>
<p>Method of adjusting p-values, by default "bonferroni". The other methods available are "BH", "holm", "hochberg", "hommel", "BY".</p>
</td>
</tr>
</table>
<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>InformativeGenes</code></td>
<td>
<p>List of informative genes selected</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>LogCPM</code></td>
<td>
<p>Log cpm data of informative genes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>DEA_Result</code></td>
<td>
<p>Differential Expression Analysis Result of informative genes</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>c(person("Md. Samir", "Farooqi", email = "ms.Farooqi@icar.gov.in", role = "aut"),
person("K.K.", "Chaturvedi", email = "kk.Chaturvedi@icar.gov.in", role = "aut"),
person("D.C.", "Mishra", email = "Dwijesh.Mishra@icar.gov.in", role = "aut"),
person("Sudhir", "Srivastava", email = "Sudhir.Srivastava@icar.gov.in", role = c("cre","aut")))
</p>


<h3>Examples</h3>

<pre><code class="language-R">  filename &lt;- system.file("extdata", "exampleData.csv", package = "TSGS")
  cdata &lt;- read.csv(filename, header = TRUE, row.names = 1, stringsAsFactors = FALSE)
  X &lt;- as.data.frame(cdata[-1,])
  y &lt;- as.numeric(cdata[1,])
  set.seed(100)
  result &lt;- featureSelect(X, y, 5, 1, 0.05, "bonferroni")
  gene_list &lt;- result$InformativeGenes
  logcpm_data &lt;- result$LogCPM
  dea_result &lt;- result$DEA_Result
</code></pre>


</div>