<div class="container">

<table style="width: 100%;"><tr>
<td>sva2</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
The adjusted sva code using irwsva.build2
</h2>

<h3>Description</h3>

<p>This function is the modified SVA function in which it uses the irwsva.build2 function rather than the irwsva.build function to build the surrogate variables. Thus, only a single line has been altered from the original SVA() function.
</p>


<h3>Usage</h3>

<pre><code class="language-R">sva2(dat, mod, mod0 = NULL, n.sv = NULL, method = c("irw", "two-step"), 
	vfilter = NULL, B = 5, numSVmethod = "be")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dat</code></td>
<td>

<p>An m by n (m cpg sites by n subjects) matrix of methylation data.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mod</code></td>
<td>

<p>A n by k model matrix corresponding to the primary model fit (see model.matrix).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mod0</code></td>
<td>

<p>A n by k0 model matrix corresponding to the null model to be compared to mod.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.sv</code></td>
<td>

<p>Optional. The number of surrogate variables to estimate, can be estimated using the num.sv function.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>

<p>Choose between the iteratively re-weighted or two-step surrogate variable estimation algorithms.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vfilter</code></td>
<td>

<p>The number of most variable CpG sites to use when building SVs, must be between 100 and m.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>B</code></td>
<td>

<p>The number of iterations of the algorithm to perform.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>numSVmethod</code></td>
<td>

<p>The method for determining the number of surrogate variables to use.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>See http://www.bioconductor.org/packages/release/bioc/manuals/sva/man/sva.pdf
</p>


<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>sv </code></td>
<td>

<p>A n by n.sv matrix where each column is a distinct surrogate variable.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pprob.gam </code></td>
<td>

<p>A vector with the posterior probability estimates that each row is affected by dependence.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pprob.b</code></td>
<td>

<p>A vector with the posterior probabiliity estimates that each row is affected by the variables in mod, but not in mod0.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.sv</code></td>
<td>

<p>The number of suggorate variables estimated.
</p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>sva Vignette http://www.biostat.jhsph.edu/~jleek/sva/
</p>


<h3>Author(s)</h3>

<p>Original sva: Jeffrey T. Leek &lt;jleek@jhsph.edu&gt;, John Storey jstorey@princeton.edu
</p>


<h3>References</h3>

<p>Original sva:
Leek JT and Storey JD. (2008) A general framework for multiple testing dependence.       Proceedings of the National Academy of Sciences, 105: 18718-18723. 
</p>
<p>Leek JT and Storey JD. (2007) Capturing heterogeneity in gene expression studies by surrogate variable analysis. PLoS Genetics, 3: e161.
</p>


</div>